<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Single workflow parameters · DoViP</title><meta name="title" content="Single workflow parameters · DoViP"/><meta property="og:title" content="Single workflow parameters · DoViP"/><meta property="twitter:title" content="Single workflow parameters · DoViP"/><meta name="description" content="Documentation for DoViP."/><meta property="og:description" content="Documentation for DoViP."/><meta property="twitter:description" content="Documentation for DoViP."/><script data-outdated-warner src="../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL=".."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../assets/documenter.js"></script><script src="../search_index.js"></script><script src="../siteinfo.js"></script><script src="../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../">DoViP</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li><a class="tocitem" href="../">Home</a></li><li><span class="tocitem">Installation</span><ul><li><a class="tocitem" href="../Installation/">Overview</a></li><li><a class="tocitem" href="../installation_conda_dependencies/">Dependencies</a></li><li><a class="tocitem" href="../installation_DoViP_env/">DoViP environment</a></li></ul></li><li><span class="tocitem">Running</span><ul><li><a class="tocitem" href="../Running/">Overview</a></li><li class="is-active"><a class="tocitem" href>Single workflow parameters</a><ul class="internal"><li><a class="tocitem" href="#General-project-arguments"><span>General project arguments</span></a></li><li><a class="tocitem" href="#ARGUMENTS-FOR-INITIAL-PREDICTION-STEPS"><span>ARGUMENTS FOR INITIAL PREDICTION STEPS</span></a></li><li><a class="tocitem" href="#ARGUMENTS-FOR-CONSENSUS-PREDICTION-STEPS"><span>ARGUMENTS FOR CONSENSUS PREDICTION STEPS</span></a></li><li><a class="tocitem" href="#NON-INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs"><span>NON-INTEGRATED viruses branch: arguments to select final viral contigs</span></a></li><li><a class="tocitem" href="#INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs"><span>INTEGRATED viruses branch: arguments to select final viral contigs</span></a></li></ul></li><li><a class="tocitem" href="../Running_multipleworkflow_params/">Multiple workflows parameters</a></li></ul></li><li><span class="tocitem">DoViP projects and outputs</span><ul><li><a class="tocitem" href="../dovip_projects/">Project definition</a></li><li><a class="tocitem" href="../dovip_outputs_overview/">Outputs - overview</a></li><li><a class="tocitem" href="../dovip_outputs_stepbystep/">Outputs - step by step breakdown</a></li><li><a class="tocitem" href="../continue_projs/">Continue projects</a></li></ul></li><li><a class="tocitem" href="../releases/">Release notes</a></li><li><a class="tocitem" href="../authors/">Authors</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li><a class="is-disabled">Running</a></li><li class="is-active"><a href>Single workflow parameters</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Single workflow parameters</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/CristinaMoraru/DoViP" title="View the repository on GitHub"><span class="docs-icon fa-brands"></span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/CristinaMoraru/DoViP/blob/main/docs_make/src/Running_singleworkflow_params.md" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="DoViP-singleworklow-parameters"><a class="docs-heading-anchor" href="#DoViP-singleworklow-parameters">DoViP singleworklow parameters</a><a id="DoViP-singleworklow-parameters-1"></a><a class="docs-heading-anchor-permalink" href="#DoViP-singleworklow-parameters" title="Permalink"></a></h1><h2 id="General-project-arguments"><a class="docs-heading-anchor" href="#General-project-arguments">General project arguments</a><a id="General-project-arguments-1"></a><a class="docs-heading-anchor-permalink" href="#General-project-arguments" title="Permalink"></a></h2><h4 id="pd_prefix"><a class="docs-heading-anchor" href="#pd_prefix"><code>pd_prefix</code></a><a id="pd_prefix-1"></a><a class="docs-heading-anchor-permalink" href="#pd_prefix" title="Permalink"></a></h4><ul><li>this is the path toward the folder where the outputs will be saved. The outputs from each individual sample will be grouped into subfolders in this folder. In the case of the <code>singleworkflow</code> mode, only one subfolder (for one sample) will be saved. In the case of the <code>multiworklow</code> mode, several subfolders will be saved, each corresponding to one sample. The subfolders will have the same names as their corresponding input fasta files (minus the file extension). This is the single parameter that should NOT be given in the .tsv file with parameters for the multiworkflow mode.     </li></ul><h4 id="inref"><a class="docs-heading-anchor" href="#inref"><code>inref</code></a><a id="inref-1"></a><a class="docs-heading-anchor-permalink" href="#inref" title="Permalink"></a></h4><ul><li>the path toward the fasta file containing the input contigs. </li><li>Accepted file extensions are: &quot;.fasta&quot;, &quot;.fna&quot;, &quot;.fst&quot;, &quot;.fas&quot;, &quot;.fa&quot;, &quot;.mfa&quot;</li><li>mandatory parameter, no default is given</li></ul><h4 id="projtype"><a class="docs-heading-anchor" href="#projtype"><code>projtype</code></a><a id="projtype-1"></a><a class="docs-heading-anchor-permalink" href="#projtype" title="Permalink"></a></h4><ul><li>the only acceptable value here is &#39;singleworkflow&#39;, and it tells DoViP that only one metagenome is analyzed in this run (or for this row, when ussing a .tsv file to send the arguments to DoViP)</li><li>mandatory parameter, no default is given</li></ul><h4 id="sample_set"><a class="docs-heading-anchor" href="#sample_set"><code>sample_set</code></a><a id="sample_set-1"></a><a class="docs-heading-anchor-permalink" href="#sample_set" title="Permalink"></a></h4><ul><li>the name of the sample/metagenome collection</li><li>mandatory parameter, no default is given</li></ul><h4 id="use_slurm"><a class="docs-heading-anchor" href="#use_slurm"><code>use_slurm</code></a><a id="use_slurm-1"></a><a class="docs-heading-anchor-permalink" href="#use_slurm" title="Permalink"></a></h4><ul><li>if calculations should be run on an High Performance Computing Cluster, using slurm and sbatch</li><li>possible values: &#39;true&#39; or &#39;false&#39;</li><li>default is &#39;false&#39;</li></ul><h4 id="continue"><a class="docs-heading-anchor" href="#continue"><code>continue</code></a><a id="continue-1"></a><a class="docs-heading-anchor-permalink" href="#continue" title="Permalink"></a></h4><ul><li>if the calculations for this sample should be continued from a previous run of the project</li><li>this is useful for example when changing the desired initial predictors (see details below) or the thresholds for the final selection of the viral contigs</li><li>possible values: &#39;true&#39; or &#39;false&#39;</li><li>default is &#39;false&#39;</li></ul><div class="admonition is-info"><header class="admonition-header">Note</header><div class="admonition-body"><p>if continue is set to &#39;false&#39; and the sample subfolder already exists, a message will appear asking to confirm or deny the deletion of the existing folder.</p></div></div><h4 id="stop_after_initial_predictors"><a class="docs-heading-anchor" href="#stop_after_initial_predictors"><code>stop_after_initial_predictors</code></a><a id="stop_after_initial_predictors-1"></a><a class="docs-heading-anchor-permalink" href="#stop_after_initial_predictors" title="Permalink"></a></h4><ul><li>if the calculations should stop after the initial predictions have finished (useful for example if the project should be continued later)</li><li>possible values: &#39;true&#39; or &#39;false&#39;</li><li>default is &#39;false&#39;</li></ul><h4 id="min_contig_length"><a class="docs-heading-anchor" href="#min_contig_length"><code>min_contig_length</code></a><a id="min_contig_length-1"></a><a class="docs-heading-anchor-permalink" href="#min_contig_length" title="Permalink"></a></h4><ul><li>the minimum length of the contigs from the input file (inref) which should be kept</li><li>only contigs equal or larger than this value will be sent to the initial predictors</li><li>value is Integer (minimum 1000)</li><li>default value is 1000</li></ul><h4 id="merge_circ_proph"><a class="docs-heading-anchor" href="#merge_circ_proph"><code>merge_circ_proph</code></a><a id="merge_circ_proph-1"></a><a class="docs-heading-anchor-permalink" href="#merge_circ_proph" title="Permalink"></a></h4><ul><li>if <code>true</code>, the two parts of prophages that wrap around circular contigs will be merged into one sequence</li><li>possible values: &#39;true&#39; (default) or &#39;false&#39;</li><li>default is &#39;true&#39;</li></ul><h2 id="ARGUMENTS-FOR-INITIAL-PREDICTION-STEPS"><a class="docs-heading-anchor" href="#ARGUMENTS-FOR-INITIAL-PREDICTION-STEPS">ARGUMENTS FOR INITIAL PREDICTION STEPS</a><a id="ARGUMENTS-FOR-INITIAL-PREDICTION-STEPS-1"></a><a class="docs-heading-anchor-permalink" href="#ARGUMENTS-FOR-INITIAL-PREDICTION-STEPS" title="Permalink"></a></h2><h3 id="geNomad-options"><a class="docs-heading-anchor" href="#geNomad-options">geNomad options</a><a id="geNomad-options-1"></a><a class="docs-heading-anchor-permalink" href="#geNomad-options" title="Permalink"></a></h3><p>If this this step is selected for running (<code>genomad_signal=do</code>), the following will happen:  </p><ul><li>DoViP will start a geNomad run in the conda environment given by the <code>genomad_env</code> parameter, using the following command:<pre><code class="nohighlight hljs">`genomad end-to-end --cleanup --splits 8 &#39;/path/to/input.fasta&#39; &#39;/path/to/output_folder&#39; `genomadDB_p` --min-score `genomad_min_score``</code></pre></li><li>DoViP will postprocess the geNomad results (file <code>_virus_summary.tsv</code>) to separate NON-INTEGRATED from INTEGRATED viruses.</li></ul><p>If, instead of running geNomad, external results should be used, the following will happen:</p><ul><li>DoViP will postprocess the geNomad results (file <code>_virus_summary.tsv</code>) to separate NON-INTEGRATED from INTEGRATED viruses.</li></ul><h4 id="genomad_signal"><a class="docs-heading-anchor" href="#genomad_signal"><code>genomad_signal</code></a><a id="genomad_signal-1"></a><a class="docs-heading-anchor-permalink" href="#genomad_signal" title="Permalink"></a></h4><ul><li><p>if geNomad predictions should be considered</p></li><li><p>possible values:</p><ul><li>&#39;do&#39; - run this predictor step. Can be always used, regardless of the signals in the previous steps</li><li>&#39;dont&#39; - don&#39;t run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a &#39;dont&#39; or &#39;remove&#39; signal in the previous step.</li><li>&#39;use&#39;- re-use predictor results calculated in previous runs of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished.</li><li>&#39;use_external&#39; - use results calculated previously by the predictor independently from a DoViP project.</li><li>&#39;ignore&#39; - doesn&#39;t take into consideration existing results calculated in a previous run of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished. </li><li>&#39;remove&#39; - removes results calculated in a previous step. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or ignore in the previous step, regardless of their &quot;progress&quot;. </li></ul></li><li><p>mandatory parameter, no default is given</p></li></ul><h4 id="genomad_res"><a class="docs-heading-anchor" href="#genomad_res"><code>genomad_res</code></a><a id="genomad_res-1"></a><a class="docs-heading-anchor-permalink" href="#genomad_res" title="Permalink"></a></h4><ul><li><code>/path/to/geNomad_output_folder</code> when the geNomad predictions were calculated outside DoViP</li><li>needed only if <code>genomad_signal=use_external</code></li><li>the path should be the exact one that was given to the geNomad command as output folder</li><li>DoViP will expect to find inside <code>/path/to/geNomad_output_folder</code> the geNomad <code>summary</code> folder (name pattern beeing composed from the <code>input_name</code>, without file extension, and <code>summary</code>):</li></ul><pre><code class="language-plaintext hljs">/path/to/geNomad_output_folder/
    ├──  input_name_summary</code></pre><h4 id="genomad_env"><a class="docs-heading-anchor" href="#genomad_env"><code>genomad_env</code></a><a id="genomad_env-1"></a><a class="docs-heading-anchor-permalink" href="#genomad_env" title="Permalink"></a></h4><ul><li>the name of the conda environment where geNomad is installed</li><li>mandatory parameter if geNomad signal is <code>do</code>, no default is given</li></ul><h4 id="genomadDB_p"><a class="docs-heading-anchor" href="#genomadDB_p"><code>genomadDB_p</code></a><a id="genomadDB_p-1"></a><a class="docs-heading-anchor-permalink" href="#genomadDB_p" title="Permalink"></a></h4><ul><li>the path toward the folder with the geNomad database</li><li>mandatory parameter if geNomad signal is <code>do</code>, no default is given</li></ul><h4 id="genomad_min_score"><a class="docs-heading-anchor" href="#genomad_min_score"><code>genomad_min_score</code></a><a id="genomad_min_score-1"></a><a class="docs-heading-anchor-permalink" href="#genomad_min_score" title="Permalink"></a></h4><ul><li>mininum score to be used by geNomad to filter viral contigs</li><li>value is Float (between 0.0 and 1.0)</li><li>default it 0.7</li></ul><h4 id="genomad_sbatch_time"><a class="docs-heading-anchor" href="#genomad_sbatch_time"><code>genomad_sbatch_time</code></a><a id="genomad_sbatch_time-1"></a><a class="docs-heading-anchor-permalink" href="#genomad_sbatch_time" title="Permalink"></a></h4><ul><li>maximum time that a sbatch job running geNomad (in slurm) is allowed to run</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given</li></ul><h4 id="genomad_cpus_per_task"><a class="docs-heading-anchor" href="#genomad_cpus_per_task"><code>genomad_cpus_per_task</code></a><a id="genomad_cpus_per_task-1"></a><a class="docs-heading-anchor-permalink" href="#genomad_cpus_per_task" title="Permalink"></a></h4><ul><li>the maximum number of CPUs in which geNomad should be run (either on a local server or on HPC).</li><li>when running on an HPC, this parameter will be further transmitted to sbatch</li><li>value is Integer</li><li>default is 20 </li></ul><h4 id="genomad_sbatch_mem"><a class="docs-heading-anchor" href="#genomad_sbatch_mem"><code>genomad_sbatch_mem</code></a><a id="genomad_sbatch_mem-1"></a><a class="docs-heading-anchor-permalink" href="#genomad_sbatch_mem" title="Permalink"></a></h4><ul><li>the maximum memory requirements if geNomad is run as an sbatch job</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given </li></ul><h3 id="DeepVir-Finder-options"><a class="docs-heading-anchor" href="#DeepVir-Finder-options">DeepVir Finder options</a><a id="DeepVir-Finder-options-1"></a><a class="docs-heading-anchor-permalink" href="#DeepVir-Finder-options" title="Permalink"></a></h3><p>If this this step is selected for running (<code>DVF_signal=do</code>), the following will happen:   </p><ul><li>DoViP will check each contig in the input file, and, if larger than the value of <code>DVF_maxContigLen</code> it will cut it in smaller pieces. This is to avoid DVF freezing when it encounters larger than 2099000.</li><li>DoViP will start a DVF run in the conda environment given by parameter <code>DVF_env</code> with the following command:  <pre><code class="nohighlight hljs">python &#39;path/to/dvf.py&#39; -i &#39;path/to/input.fasta&#39; -o&#39; path/to/output/folder&#39; -l &#39;min_contig_len&#39; -c `DVF_cpus_per_task`</code></pre></li><li>DoViP will merge back into a single entry the larger contigs which have been split in the first step, with the score and the p-value repersentig the median of the smaller contigs.</li><li>DoViP will postprocess all DVF results to remove all contigs with a score smaller than <code>DVF_scoreTh</code> and a p-value larger than <code>DVF_pThreshold</code>. Only NON-INTEGRATED viruses are outputted by DVF.</li></ul><h4 id="DVF_signal"><a class="docs-heading-anchor" href="#DVF_signal"><code>DVF_signal</code></a><a id="DVF_signal-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_signal" title="Permalink"></a></h4><ul><li><p>if DeepVir Finder predictions should be considered</p></li><li><p>possible values:</p><ul><li>&#39;do&#39; - run this predictor step. Can be always used, regardless of the signals in the previous steps</li><li>&#39;dont&#39; - don&#39;t run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a &#39;dont&#39; or &#39;remove&#39; signal in the previous step.</li><li>&#39;use&#39;- re-use predictor results calculated in previous runs of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished.</li><li>&#39;use_external&#39; - use results calculated previously by the predictor independently from a DoViP project.</li><li>&#39;ignore&#39; - doesn&#39;t take into consideration existing results calculated in a previous run of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished. </li><li>&#39;remove&#39; - removes results calculated in a previous step. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or ignore in the previous step, regardless of their &quot;progress&quot;. </li></ul></li><li><p>mandatory parameter if DVF should be run, no default is given</p></li></ul><h4 id="DVF_env"><a class="docs-heading-anchor" href="#DVF_env"><code>DVF_env</code></a><a id="DVF_env-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_env" title="Permalink"></a></h4><ul><li>the name of the conda environment where DVF is installed</li><li>mandatory parameter, no default is given    </li></ul><h4 id="DVF_maxContigLen"><a class="docs-heading-anchor" href="#DVF_maxContigLen"><code>DVF_maxContigLen</code></a><a id="DVF_maxContigLen-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_maxContigLen" title="Permalink"></a></h4><ul><li>the maximum length that a contig should have for DVF. If larger than this value, it will be cut into smaller pieces before being inputted into DVF.</li><li>value is Integer</li><li>default is 2099000</li></ul><h4 id="DVF_scoreTh"><a class="docs-heading-anchor" href="#DVF_scoreTh"><code>DVF_scoreTh</code></a><a id="DVF_scoreTh-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_scoreTh" title="Permalink"></a></h4><ul><li>minimum score value calculated by DVF so that contigs are kept by DoViP </li><li>value is Float (between 0 and 1)</li><li>default is 0.7</li></ul><h4 id="DVF_pThreshold"><a class="docs-heading-anchor" href="#DVF_pThreshold"><code>DVF_pThreshold</code></a><a id="DVF_pThreshold-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_pThreshold" title="Permalink"></a></h4><ul><li>the maximum p-value calculated by DVF so that contigs are kept by DoViP </li><li>value is Float (between 0 and 1)</li><li>default is 0.01</li></ul><h4 id="DVF_p"><a class="docs-heading-anchor" href="#DVF_p"><code>DVF_p</code></a><a id="DVF_p-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_p" title="Permalink"></a></h4><ul><li>path/to/dvf.py  </li><li>mandatory parameter if DVF should be run, no default is given   </li></ul><h4 id="DVF_sbatch_time"><a class="docs-heading-anchor" href="#DVF_sbatch_time"><code>DVF_sbatch_time</code></a><a id="DVF_sbatch_time-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_sbatch_time" title="Permalink"></a></h4><ul><li>maximum time that a sbatch job running geNomad (in slurm) is allowed to run</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given</li></ul><h4 id="DVF_cpus_per_task"><a class="docs-heading-anchor" href="#DVF_cpus_per_task"><code>DVF_cpus_per_task</code></a><a id="DVF_cpus_per_task-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_cpus_per_task" title="Permalink"></a></h4><ul><li>the maximum number of CPUs in which DVF should be run (either on a local server or on HPC)</li><li>when running on an HPC, this parameter will be further transmitted to sbatch</li><li>value is Integer</li><li>default is 15 </li></ul><h4 id="DVF_sbatch_mem"><a class="docs-heading-anchor" href="#DVF_sbatch_mem"><code>DVF_sbatch_mem</code></a><a id="DVF_sbatch_mem-1"></a><a class="docs-heading-anchor-permalink" href="#DVF_sbatch_mem" title="Permalink"></a></h4><ul><li>the maximum memmory requirements if DVF is run as an sbatch job</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given        </li></ul><h3 id="VirSorter2-options"><a class="docs-heading-anchor" href="#VirSorter2-options">VirSorter2 options</a><a id="VirSorter2-options-1"></a><a class="docs-heading-anchor-permalink" href="#VirSorter2-options" title="Permalink"></a></h3><p>If this this step is selected for running (<code>virSorter2_signal=do</code>), the following will happen:  </p><ul><li><p>DoViP will start a VirSorter2 run in the conda environment given by the <code>virSorter2_env</code> parameter, using the following command:</p><pre><code class="nohighlight hljs">virsorter run -i &#39;/path/to/input.fasta&#39; -w &#39;path/to/output/folder&#39; -j `virSorter2_cpus_per_task` -d `virSorter2DB_p` --min-length `min_contig_length` --min-score `virSorter2_min_score` --tmpdir &#39;path/to/output/folder/tmp&#39; --viral-gene-required --exclude-lt2gene --hallmark-required-on-short$(hc) --include-groups dsDNAphage,NCLDV,RNA,ssDNA,lavidaviridae --keep-original-seq --rm-tmpdir

# the value of $(hc) in the code above is &quot; --high-confidence-only&quot; if the DoViP parameter `virSorter2_high_confidence_only=true` or &quot;&quot; if `virSorter2_high_confidence_only=false`</code></pre></li><li><p>DoViP will postprocess the VirSorter2 results (<code>final-viral-score.tsv</code> and <code>final-viral-boundary.tsv</code> files) to separate NON-INTEGRATED from INTEGRATED viruses.</p></li></ul><h4 id="virSorter2_signal"><a class="docs-heading-anchor" href="#virSorter2_signal"><code>virSorter2_signal</code></a><a id="virSorter2_signal-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2_signal" title="Permalink"></a></h4><ul><li><p>if VirSorter2 predictions should be considered</p></li><li><p>possible values:</p><ul><li>&#39;do&#39; - run this predictor step. Can be always used, regardless of the signals in the previous steps</li><li>&#39;dont&#39; - don&#39;t run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a &#39;dont&#39; or &#39;remove&#39; signal in the previous step.</li><li>&#39;use&#39;- re-use predictor results calculated in previous runs of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished.</li><li>&#39;use_external&#39; - use results calculated previously by the predictor independently from a DoViP project.</li><li>&#39;ignore&#39; - doesn&#39;t take into consideration existing results calculated in a previous run of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished. </li><li>&#39;remove&#39; - removes results calculated in a previous step. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or ignore in the previous step, regardless of their &quot;progress&quot;. </li></ul></li><li><p>mandatory parameter, no default is given</p></li></ul><h4 id="virSorter2_res"><a class="docs-heading-anchor" href="#virSorter2_res"><code>virSorter2_res</code></a><a id="virSorter2_res-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2_res" title="Permalink"></a></h4><ul><li><code>path/to/VirSorter2</code><em>output</em>folder when VirSorter2 results were calculated outside DoViP</li><li>needed only if <code>virSorter2_signal=use_external</code></li><li>the path should be the exact one that was given to the VirSorter2 command as output folder</li><li>DoViP expects to find inside <code>path/to/VirSorter2_output_folder</code> the following files:</li></ul><pre><code class="language-plaintext hljs">path/to/VirSorter2_output_folder/
    ├── final-viral-score.tsv 
    ├── final-viral-boundary.tsv</code></pre><h4 id="virSorter2_env"><a class="docs-heading-anchor" href="#virSorter2_env"><code>virSorter2_env</code></a><a id="virSorter2_env-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2_env" title="Permalink"></a></h4><ul><li>the name of the conda environment where VirSorter2 is installed</li><li>mandatory parameter if virSorter2 signal is <code>do</code>, no default is given, no default is given</li></ul><h4 id="virSorter2DB_p"><a class="docs-heading-anchor" href="#virSorter2DB_p"><code>virSorter2DB_p</code></a><a id="virSorter2DB_p-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2DB_p" title="Permalink"></a></h4><ul><li>the path toward the folder with the virSorter2 database</li><li>mandatory parameter if virSorter2 signal is <code>do</code>, no default is given, no default is given  </li></ul><h4 id="virSorter2_high_confidence_only"><a class="docs-heading-anchor" href="#virSorter2_high_confidence_only"><code>virSorter2_high_confidence_only</code></a><a id="virSorter2_high_confidence_only-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2_high_confidence_only" title="Permalink"></a></h4><ul><li>will tell VirSorter2 if it should keep only the high confidence contigs</li><li>possible values: &#39;true&#39; or &#39;false&#39;</li><li>default is &#39;false&#39;</li></ul><h4 id="virSorter2_min_score"><a class="docs-heading-anchor" href="#virSorter2_min_score"><code>virSorter2_min_score</code></a><a id="virSorter2_min_score-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2_min_score" title="Permalink"></a></h4><ul><li>minimum score to be used by VirSorter2 to filter viral contigs</li><li>value is Float (between 0.0 and 1.0)</li><li>default it 0.5</li></ul><h4 id="virSorter2_sbatch_time"><a class="docs-heading-anchor" href="#virSorter2_sbatch_time"><code>virSorter2_sbatch_time</code></a><a id="virSorter2_sbatch_time-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2_sbatch_time" title="Permalink"></a></h4><ul><li>maximum time that a sbatch job running VirSorter2 (in slurm) is allowed to run</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given</li></ul><h4 id="virSorter2_cpus_per_task"><a class="docs-heading-anchor" href="#virSorter2_cpus_per_task"><code>virSorter2_cpus_per_task</code></a><a id="virSorter2_cpus_per_task-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2_cpus_per_task" title="Permalink"></a></h4><ul><li>the maximum number of CPUs in which VirSorter2 should be run (either on a local server or on HPC)</li><li>when running on an HPC, this parameter will be further transmited to sbatch</li><li>value is Integer</li><li>default is 20 </li></ul><h4 id="virSorter2_sbatch_mem"><a class="docs-heading-anchor" href="#virSorter2_sbatch_mem"><code>virSorter2_sbatch_mem</code></a><a id="virSorter2_sbatch_mem-1"></a><a class="docs-heading-anchor-permalink" href="#virSorter2_sbatch_mem" title="Permalink"></a></h4><ul><li>the maximum memory requirements if VirSorter2 is run as an sbatch job</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given </li></ul><h3 id="VIBRANT-options"><a class="docs-heading-anchor" href="#VIBRANT-options">VIBRANT options</a><a id="VIBRANT-options-1"></a><a class="docs-heading-anchor-permalink" href="#VIBRANT-options" title="Permalink"></a></h3><p>If this this step is selected for running (<code>vibrant_signal=do</code>), the following will happen:  </p><ul><li>DoViP will start a VIBRANT run in the conda environment given by the <code>vibrant_env</code> parameter, using the following command:<pre><code class="nohighlight hljs">python3 `vibrant_p` -f nucl -i &#39;/path/to/input.fasta&#39; -folder &#39;path/to/output/folder&#39; -t `vibrant_cpus_per_task` -d `vibrantDB_p` -l `min_contig_length`</code></pre></li><li>DoViP will postprocess the VIBRANT outputs (<code>VIBRANT_integrated_prophage_coordinates_$(sampleName).tsv</code>, <code>$(sampleName).phages_lytic.fna</code>) to separate NON-INTEGRATED from INTEGRATED viruses.</li></ul><h4 id="vibrant_signal"><a class="docs-heading-anchor" href="#vibrant_signal"><code>vibrant_signal</code></a><a id="vibrant_signal-1"></a><a class="docs-heading-anchor-permalink" href="#vibrant_signal" title="Permalink"></a></h4><ul><li><p>if VIBRANT predictions should be considered</p></li><li><p>possible values:</p><ul><li>&#39;do&#39; - run this predictor step. Can be always used, regardless of the signals in the previous steps</li><li>&#39;dont&#39; - don&#39;t run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a &#39;dont&#39; or &#39;remove&#39; signal in the previous step.</li><li>&#39;use&#39;- re-use predictor results calculated in previous runs of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished.</li><li>&#39;use_external&#39; - use results calculated previously by the predictor independently from a DoViP project.</li><li>&#39;ignore&#39; - doesn&#39;t take into consideration existing results calculated in a previous run of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished. </li><li>&#39;remove&#39; - removes results calculated in a previous step. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or ignore in the previous step, regardless of their &quot;progress&quot;. </li></ul></li><li><p>mandatory parameter, no default is given</p></li></ul><h4 id="vibrant_res"><a class="docs-heading-anchor" href="#vibrant_res"><code>vibrant_res</code></a><a id="vibrant_res-1"></a><a class="docs-heading-anchor-permalink" href="#vibrant_res" title="Permalink"></a></h4><ul><li><code>path/to/VIBRANT_output_folder</code> when the VIBRANT results were calculated outside DoViP</li><li>needed only if <code>vibrant_signal=use_external</code></li><li>the path should be the exact one that was given to the VIBRANT command as output folder</li><li>DoViP expects to find inside <code>path/to/VIBRANT_output_folder</code> another folder, created by VIBRANT itself, having as prefix <code>VIBRANT</code> and as suffix the <code>input_name</code> (without extension):</li></ul><pre><code class="language-plaintext hljs">path/to/VIBRANT_output_folder/
    ├── VIBRANT_input_name </code></pre><h4 id="vibrant_env"><a class="docs-heading-anchor" href="#vibrant_env"><code>vibrant_env</code></a><a id="vibrant_env-1"></a><a class="docs-heading-anchor-permalink" href="#vibrant_env" title="Permalink"></a></h4><ul><li>the name of the conda environment where VIBRANT is installed</li><li>mandatory parameter if VIBRANT signal is <code>do</code>, no default is given, no default is given</li></ul><h4 id="vibrant_p"><a class="docs-heading-anchor" href="#vibrant_p"><code>vibrant_p</code></a><a id="vibrant_p-1"></a><a class="docs-heading-anchor-permalink" href="#vibrant_p" title="Permalink"></a></h4><ul><li>path/to/VIBRANT_run.py  </li><li>mandatory parameter if VIBRANT signal is <code>do</code>, no default is given, no default is given</li></ul><h4 id="vibrantDB_p"><a class="docs-heading-anchor" href="#vibrantDB_p"><code>vibrantDB_p</code></a><a id="vibrantDB_p-1"></a><a class="docs-heading-anchor-permalink" href="#vibrantDB_p" title="Permalink"></a></h4><ul><li>the path toward the folder with the VIBRANT database</li><li>mandatory parameter if VIBRANT signal is <code>do</code>, no default is given, no default is given </li></ul><h4 id="vibrant_sbatch_time"><a class="docs-heading-anchor" href="#vibrant_sbatch_time"><code>vibrant_sbatch_time</code></a><a id="vibrant_sbatch_time-1"></a><a class="docs-heading-anchor-permalink" href="#vibrant_sbatch_time" title="Permalink"></a></h4><ul><li>maximum time that a sbatch job running VIBRANT (in slurm) is allowed to run</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC and VIBRANT signal is <code>do</code>, no default is given </li></ul><h4 id="vibrant_cpus_per_task"><a class="docs-heading-anchor" href="#vibrant_cpus_per_task"><code>vibrant_cpus_per_task</code></a><a id="vibrant_cpus_per_task-1"></a><a class="docs-heading-anchor-permalink" href="#vibrant_cpus_per_task" title="Permalink"></a></h4><ul><li>the maximum number of CPUs in which VIBRANT should be run (either on a local server or on HPC)</li><li>when running on an HPC, this parameter will be further transmitted to sbatch</li><li>value is Integer</li><li>default is 20 </li></ul><h4 id="vibrant_sbatch_mem"><a class="docs-heading-anchor" href="#vibrant_sbatch_mem"><code>vibrant_sbatch_mem</code></a><a id="vibrant_sbatch_mem-1"></a><a class="docs-heading-anchor-permalink" href="#vibrant_sbatch_mem" title="Permalink"></a></h4><ul><li>the maximum memmory requirements if VIBRANT is run as an sbatch job</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC and VIBRANT signal is <code>do</code>, no default is given </li></ul><h3 id="ViralVerify-options"><a class="docs-heading-anchor" href="#ViralVerify-options">ViralVerify options</a><a id="ViralVerify-options-1"></a><a class="docs-heading-anchor-permalink" href="#ViralVerify-options" title="Permalink"></a></h3><p>If this this step is selected for running (<code>viralVerify_signal=do</code>), the following will happen:  </p><ul><li>DoViP will start a ViralVerify run in the conda environment given by the <code>viralVerify_env</code> parameter, using the following command:<pre><code class="nohighlight hljs">viralVerify_p` -f &#39;/path/to/input.fasta&#39; -o &#39;path/to/output/folder&#39; --hmm `viralVerifyDB_p` --thr `viralVerfify_threshold` -t `viralVerify_cpus_per_task` -p</code></pre></li><li>DoViP will postprocess ViralVerify outputs (<code>_result_table.csv</code>) to select only the viral contigs with a score bigger than <code>viralVerfify_threshold</code>.  Only NON-INTEGRATED viruses are outputted by ViralVerify.</li></ul><h4 id="viralVerify_signal"><a class="docs-heading-anchor" href="#viralVerify_signal"><code>viralVerify_signal</code></a><a id="viralVerify_signal-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerify_signal" title="Permalink"></a></h4><ul><li><p>if ViralVerify predictions should be considered</p></li><li><p>possible values:</p><ul><li>&#39;do&#39; - run this predictor step. Can be always used, regardless of the signals in the previous steps</li><li>&#39;dont&#39; - don&#39;t run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a &#39;dont&#39; or &#39;remove&#39; signal in the previous step.</li><li>&#39;use&#39;- re-use predictor results calculated in previous runs of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished.</li><li>&#39;use_external&#39; - use results calculated previously by the predictor independently from a DoViP project.</li><li>&#39;ignore&#39; - doesn&#39;t take into consideration existing results calculated in a previous run of this project. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or &#39;ignore&#39; in the previous step, and, if their respective &quot;progress&quot; is finished. </li><li>&#39;remove&#39; - removes results calculated in a previous step. It can&#39;t be used if the project is new (continue = false). For continued projects, it can be used only after &#39;do&#39;, &#39;use&#39; or ignore in the previous step, regardless of their &quot;progress&quot;. </li></ul></li><li><p>mandatory parameter, no default is given</p></li></ul><h4 id="viralVerify_res"><a class="docs-heading-anchor" href="#viralVerify_res"><code>viralVerify_res</code></a><a id="viralVerify_res-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerify_res" title="Permalink"></a></h4><ul><li><code>path/to/ViralVerify_output_folder</code> when the ViralVerify results were calculated outside DoViP</li><li>needed only if <code>viralVerify_signal=use_external</code></li><li>the path should be the exact one that was given to the ViralVerify command as output folder</li><li>DoViP expects to find inside <code>path/to/ViralVerify_output_folder</code> a file created by ViralVerify having as prefix the <code>input_name</code> (without extension) and as suffix <code>result_table.csv</code>:</li></ul><pre><code class="language-plaintext hljs">path/to/ViralVerify_output_folder/
    ├── input_name_result_table.csv</code></pre><h4 id="viralVerify_env"><a class="docs-heading-anchor" href="#viralVerify_env"><code>viralVerify_env</code></a><a id="viralVerify_env-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerify_env" title="Permalink"></a></h4><ul><li>the name of the conda environment where ViralVerify is installed</li><li>mandatory parameter if ViralVerify signal is <code>do</code>, no default is given, no default is given</li></ul><h4 id="viralVerifyDB_p"><a class="docs-heading-anchor" href="#viralVerifyDB_p"><code>viralVerifyDB_p</code></a><a id="viralVerifyDB_p-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerifyDB_p" title="Permalink"></a></h4><ul><li>the path toward the HMM profile file of the ViralVerify database</li><li>mandatory parameter if ViralVerify signal is <code>do</code>, no default is given, no default is given</li></ul><h4 id="viralVerify_p"><a class="docs-heading-anchor" href="#viralVerify_p"><code>viralVerify_p</code></a><a id="viralVerify_p-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerify_p" title="Permalink"></a></h4><ul><li>path/to/bin/viralverify </li><li>mandatory parameter if ViralVerify signal is <code>do</code>, no default is given, no default is given</li></ul><h4 id="viralVerfify_threshold"><a class="docs-heading-anchor" href="#viralVerfify_threshold"><code>viralVerfify_threshold</code></a><a id="viralVerfify_threshold-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerfify_threshold" title="Permalink"></a></h4><ul><li>score used to select viral contigs outputted by ViralVerify. </li><li>value is Integer (between 1 and 1000)</li><li>default is 7</li></ul><h4 id="viralVerify_sbatch_time"><a class="docs-heading-anchor" href="#viralVerify_sbatch_time"><code>viralVerify_sbatch_time</code></a><a id="viralVerify_sbatch_time-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerify_sbatch_time" title="Permalink"></a></h4><ul><li>maximum time that a sbatch job running ViralVeriffy (in slurm) is allowed to run</li><li>mandatory parameter when running on HPC and ViralVerify signal is <code>do</code>, no default is given</li></ul><h4 id="viralVerify_cpus_per_task"><a class="docs-heading-anchor" href="#viralVerify_cpus_per_task"><code>viralVerify_cpus_per_task</code></a><a id="viralVerify_cpus_per_task-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerify_cpus_per_task" title="Permalink"></a></h4><ul><li>the maximum number of CPUs in which ViralVerify should be run (either on a local server or on HPC)</li><li>value is Integer</li><li>default is 20 </li></ul><h4 id="viralVerify_sbatch_mem"><a class="docs-heading-anchor" href="#viralVerify_sbatch_mem"><code>viralVerify_sbatch_mem</code></a><a id="viralVerify_sbatch_mem-1"></a><a class="docs-heading-anchor-permalink" href="#viralVerify_sbatch_mem" title="Permalink"></a></h4><ul><li>the maximum memmory requirements if ViralVerify is run as an sbatch job</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC and ViralVerify signal is <code>do</code>, no default is given </li></ul><h2 id="ARGUMENTS-FOR-CONSENSUS-PREDICTION-STEPS"><a class="docs-heading-anchor" href="#ARGUMENTS-FOR-CONSENSUS-PREDICTION-STEPS">ARGUMENTS FOR CONSENSUS PREDICTION STEPS</a><a id="ARGUMENTS-FOR-CONSENSUS-PREDICTION-STEPS-1"></a><a class="docs-heading-anchor-permalink" href="#ARGUMENTS-FOR-CONSENSUS-PREDICTION-STEPS" title="Permalink"></a></h2><h3 id="CheckV-arguments"><a class="docs-heading-anchor" href="#CheckV-arguments">CheckV arguments</a><a id="CheckV-arguments-1"></a><a class="docs-heading-anchor-permalink" href="#CheckV-arguments" title="Permalink"></a></h3><p>CheckV is used both on the NON-INTEGRATED and INTEGRATED viruses branch to detect viral contigs with host contamination (if they are on the NON-INTEGRATED branch they will be moved to the INTEGRATED branch) and remove the host regions.   In all cases, CheckV is run with the following command:  </p><pre><code class="nohighlight hljs">checkv end_to_end &#39;path/to/input.fasta&#39; &#39;path/to/output_folder&#39; -t `checkv_cpus_per_task` -d `checkvDB_p` --remove_tmp</code></pre><p>The CheckV statistics will be incorporated in the final virus contig tables.</p><h4 id="checkv_env"><a class="docs-heading-anchor" href="#checkv_env"><code>checkv_env</code></a><a id="checkv_env-1"></a><a class="docs-heading-anchor-permalink" href="#checkv_env" title="Permalink"></a></h4><ul><li>the name of the conda environment where CheckV is installed</li><li>mandatory parameter, no default is given</li></ul><h4 id="checkvDB_p"><a class="docs-heading-anchor" href="#checkvDB_p"><code>checkvDB_p</code></a><a id="checkvDB_p-1"></a><a class="docs-heading-anchor-permalink" href="#checkvDB_p" title="Permalink"></a></h4><ul><li>the path toward the folder with the CheckV database</li><li>mandatory parameter, no default is given</li></ul><h4 id="checkv_sbatch_time"><a class="docs-heading-anchor" href="#checkv_sbatch_time"><code>checkv_sbatch_time</code></a><a id="checkv_sbatch_time-1"></a><a class="docs-heading-anchor-permalink" href="#checkv_sbatch_time" title="Permalink"></a></h4><ul><li>maximum time that a sbatch job running CheckV (in slurm) is allowed to run</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given</li></ul><h4 id="checkv_cpus_per_task"><a class="docs-heading-anchor" href="#checkv_cpus_per_task"><code>checkv_cpus_per_task</code></a><a id="checkv_cpus_per_task-1"></a><a class="docs-heading-anchor-permalink" href="#checkv_cpus_per_task" title="Permalink"></a></h4><ul><li>the maximum number of CPUs in which CheckV should be run (either on a local server or on HPC)</li><li>when running on an HPC, this parameter will be further transmitted to sbatch</li><li>value is Integer</li><li>default is 2 </li></ul><h4 id="checkv_sbatch_mem"><a class="docs-heading-anchor" href="#checkv_sbatch_mem"><code>checkv_sbatch_mem</code></a><a id="checkv_sbatch_mem-1"></a><a class="docs-heading-anchor-permalink" href="#checkv_sbatch_mem" title="Permalink"></a></h4><ul><li>the maximum memmory requirements if checkV is run as an sbatch job</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given </li></ul><h3 id="PhaTYP-arguments"><a class="docs-heading-anchor" href="#PhaTYP-arguments">PhaTYP arguments</a><a id="PhaTYP-arguments-1"></a><a class="docs-heading-anchor-permalink" href="#PhaTYP-arguments" title="Permalink"></a></h3><p>PhaTYP is used on the NON-INTEGRATED viruses branch to predict the potential lifestyle of the respective viruses.   PhaTYP is run with the following command:</p><pre><code class="nohighlight hljs">python `phaTYP_p/PhaTYP_single.py` --contigs &#39;path/to/input.fasta&#39; --threads `phaTYP_cpus_per_task` --len `min_contig_length` --rootpth &#39;path/to/temp_output_folder&#39; --out &#39;path/to/output_folder&#39; --dbdir `phaTYPdb_p` --parampth `phaTYPparam_p`</code></pre><p>The PhaTYP predictions will be incorporated in the final virus contig tables.</p><h4 id="phaTYP_env"><a class="docs-heading-anchor" href="#phaTYP_env"><code>phaTYP_env</code></a><a id="phaTYP_env-1"></a><a class="docs-heading-anchor-permalink" href="#phaTYP_env" title="Permalink"></a></h4><ul><li>the name of the conda environment where PhaTYP is installed</li><li>mandatory parameter, no default is given</li></ul><h4 id="phaTYP_p"><a class="docs-heading-anchor" href="#phaTYP_p"><code>phaTYP_p</code></a><a id="phaTYP_p-1"></a><a class="docs-heading-anchor-permalink" href="#phaTYP_p" title="Permalink"></a></h4><ul><li>the path toward the folder with PhaTYP</li><li>mandatory parameter, no default is given</li></ul><h4 id="phaTYPdb_p"><a class="docs-heading-anchor" href="#phaTYPdb_p"><code>phaTYPdb_p</code></a><a id="phaTYPdb_p-1"></a><a class="docs-heading-anchor-permalink" href="#phaTYPdb_p" title="Permalink"></a></h4><ul><li>the path toward the folder with the PhaTYP database</li><li>mandatory parameter, no default is given</li></ul><h4 id="phaTYPparam_p"><a class="docs-heading-anchor" href="#phaTYPparam_p"><code>phaTYPparam_p</code></a><a id="phaTYPparam_p-1"></a><a class="docs-heading-anchor-permalink" href="#phaTYPparam_p" title="Permalink"></a></h4><ul><li>the path toward the folder with the PhaTYP parameters (see PhaTYP manual)</li><li>mandatory parameter, no default is given</li></ul><h4 id="phaTYP_sbatch_time"><a class="docs-heading-anchor" href="#phaTYP_sbatch_time"><code>phaTYP_sbatch_time</code></a><a id="phaTYP_sbatch_time-1"></a><a class="docs-heading-anchor-permalink" href="#phaTYP_sbatch_time" title="Permalink"></a></h4><ul><li>maximum time that a sbatch job running PhaTYP (in slurm) is allowed to run</li><li>this parameter will be further transmited to sbatch</li><li>mandatory parameter when running on HPC, no default is given</li></ul><h4 id="phaTYP_cpus_per_task"><a class="docs-heading-anchor" href="#phaTYP_cpus_per_task"><code>phaTYP_cpus_per_task</code></a><a id="phaTYP_cpus_per_task-1"></a><a class="docs-heading-anchor-permalink" href="#phaTYP_cpus_per_task" title="Permalink"></a></h4><ul><li>the maximum number of CPUs in which PhaTYP should be run (either on a local server or on HPC)</li><li>when running on an HPC, this parameter will be further transmitted to sbatch</li><li>value is Integer</li><li>default is 2 </li></ul><h4 id="phaTYP_sbatch_mem"><a class="docs-heading-anchor" href="#phaTYP_sbatch_mem"><code>phaTYP_sbatch_mem</code></a><a id="phaTYP_sbatch_mem-1"></a><a class="docs-heading-anchor-permalink" href="#phaTYP_sbatch_mem" title="Permalink"></a></h4><ul><li>the maximum memory requirements if PhaTYP is run as an sbatch job</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given </li></ul><h3 id="GeNomad-for-taxonomy-arguments"><a class="docs-heading-anchor" href="#GeNomad-for-taxonomy-arguments">GeNomad for taxonomy arguments</a><a id="GeNomad-for-taxonomy-arguments-1"></a><a class="docs-heading-anchor-permalink" href="#GeNomad-for-taxonomy-arguments" title="Permalink"></a></h3><h4 id="genomadtax_sbatch_time"><a class="docs-heading-anchor" href="#genomadtax_sbatch_time"><code>genomadtax_sbatch_time</code></a><a id="genomadtax_sbatch_time-1"></a><a class="docs-heading-anchor-permalink" href="#genomadtax_sbatch_time" title="Permalink"></a></h4><ul><li>maximum time that a sbatch job running geNomad for taxonomy (in slurm) is allowed to run</li><li>this parameter will be further transmited to sbatch</li><li>mandatory parameter when running on HPC, no default is given</li></ul><h4 id="genomadtax_cpus_per_task"><a class="docs-heading-anchor" href="#genomadtax_cpus_per_task"><code>genomadtax_cpus_per_task</code></a><a id="genomadtax_cpus_per_task-1"></a><a class="docs-heading-anchor-permalink" href="#genomadtax_cpus_per_task" title="Permalink"></a></h4><ul><li>the maximum number of CPUs in which geNomad for taxonomy should be run (either on a local server or on HPC)</li><li>when running on an HPC, this parameter will be further transmitted to sbatch</li><li>value is Integer</li><li>default is 20 </li></ul><h4 id="genomadtax_sbatch_mem"><a class="docs-heading-anchor" href="#genomadtax_sbatch_mem"><code>genomadtax_sbatch_mem</code></a><a id="genomadtax_sbatch_mem-1"></a><a class="docs-heading-anchor-permalink" href="#genomadtax_sbatch_mem" title="Permalink"></a></h4><ul><li>the maximum memory requirements if geNomad for taxonomy is run as an sbatch job</li><li>this parameter will be further transmitted to sbatch</li><li>mandatory parameter when running on HPC, no default is given </li></ul><h2 id="NON-INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs"><a class="docs-heading-anchor" href="#NON-INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs">NON-INTEGRATED viruses branch: arguments to select final viral contigs</a><a id="NON-INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs-1"></a><a class="docs-heading-anchor-permalink" href="#NON-INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs" title="Permalink"></a></h2><p>At the end of the workflow DoViP keeps only the NON-INTEGRATED viral contigs that fulfill certain criteria. The selection depends on the method used by CheckV to estimate the viral contig completeness, on the value of the viral contig completeness, and on the initial predictor coverage value (average number of predictors at each base along the viral sequence) for the respective viral contig.  </p><h3 id="If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:"><a class="docs-heading-anchor" href="#If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:">If CheckV could not estimate the completeness of the viral contigs:</a><a id="If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:-1"></a><a class="docs-heading-anchor-permalink" href="#If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:" title="Permalink"></a></h3><ul><li>it means that CheckV could not find similarities in its database</li><li>than only the number of viral predictors is taken into consideration. </li><li>I recommend using a higher number of initial predictors, to increase the confidence that the respective contig is actually viral  </li></ul><h4 id="NONInt_th_num_predictors_CheckV_NA"><a class="docs-heading-anchor" href="#NONInt_th_num_predictors_CheckV_NA"><code>NONInt_th_num_predictors_CheckV_NA</code></a><a id="NONInt_th_num_predictors_CheckV_NA-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_num_predictors_CheckV_NA" title="Permalink"></a></h4><ul><li>this parameters gives the minimum number of initial predictors necessary to keep a viral contig if completeness is NA</li><li>value is Integer</li><li>default is 3</li></ul><h3 id="If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:"><a class="docs-heading-anchor" href="#If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:">If CheckV used AAIHighConf method to estimate the completeness of the viral contigs:</a><a id="If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:-1"></a><a class="docs-heading-anchor-permalink" href="#If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:" title="Permalink"></a></h3><ul><li>it means that it found related viruses in its database and it can estimate the completeness with high precision</li><li>all viral contigs with a completeness of at least 90% are kept, regardless of the number of initial predictors </li></ul><h4 id="NONInt_th_num_predictors_CheckV_AAIHighConf"><a class="docs-heading-anchor" href="#NONInt_th_num_predictors_CheckV_AAIHighConf"><code>NONInt_th_num_predictors_CheckV_AAIHighConf</code></a><a id="NONInt_th_num_predictors_CheckV_AAIHighConf-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_num_predictors_CheckV_AAIHighConf" title="Permalink"></a></h4><ul><li>this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is AAIHighConf</li><li>value is Integer</li><li>default is 1</li></ul><h4 id="NONInt_th_completeness_CheckV_AAIHighConf"><a class="docs-heading-anchor" href="#NONInt_th_completeness_CheckV_AAIHighConf"><code>NONInt_th_completeness_CheckV_AAIHighConf</code></a><a id="NONInt_th_completeness_CheckV_AAIHighConf-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_completeness_CheckV_AAIHighConf" title="Permalink"></a></h4><ul><li>this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is AAIHighConf</li><li>value is Float</li><li>default is 30.0</li></ul><h3 id="If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:"><a class="docs-heading-anchor" href="#If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:">If CheckV used AAIMediumConf method to estimate the completeness of the viral contigs:</a><a id="If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:-1"></a><a class="docs-heading-anchor-permalink" href="#If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:" title="Permalink"></a></h3><ul><li>it means that it found related viruses in its database and it can estimate the completeness with moderate precission</li></ul><h4 id="NONInt_th_num_predictors_CheckV_AAIMediumConf"><a class="docs-heading-anchor" href="#NONInt_th_num_predictors_CheckV_AAIMediumConf"><code>NONInt_th_num_predictors_CheckV_AAIMediumConf</code></a><a id="NONInt_th_num_predictors_CheckV_AAIMediumConf-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_num_predictors_CheckV_AAIMediumConf" title="Permalink"></a></h4><ul><li>this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is AAIMediumConf</li><li>value is Integer</li><li>default is 2    </li></ul><h4 id="NONInt_th_completeness_CheckV_AAIMediumConf"><a class="docs-heading-anchor" href="#NONInt_th_completeness_CheckV_AAIMediumConf"><code>NONInt_th_completeness_CheckV_AAIMediumConf</code></a><a id="NONInt_th_completeness_CheckV_AAIMediumConf-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_completeness_CheckV_AAIMediumConf" title="Permalink"></a></h4><ul><li>this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is AAIMediumConf</li><li>value is Float</li><li>default is 10.0</li></ul><h3 id="If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:"><a class="docs-heading-anchor" href="#If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:">If CheckV used HMM method to estimate the completeness of the viral contigs:</a><a id="If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:-1"></a><a class="docs-heading-anchor-permalink" href="#If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:" title="Permalink"></a></h3><ul><li>it means that it found distantly related viruses in its database and it can estimate the completeness with low precision</li></ul><h4 id="NONInt_th_num_predictors_CheckV_HMM"><a class="docs-heading-anchor" href="#NONInt_th_num_predictors_CheckV_HMM"><code>NONInt_th_num_predictors_CheckV_HMM</code></a><a id="NONInt_th_num_predictors_CheckV_HMM-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_num_predictors_CheckV_HMM" title="Permalink"></a></h4><ul><li>this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is HMM</li><li>value is Integer</li><li>default is 2    </li></ul><h4 id="NONInt_th_completeness_CheckV_HMM"><a class="docs-heading-anchor" href="#NONInt_th_completeness_CheckV_HMM"><code>NONInt_th_completeness_CheckV_HMM</code></a><a id="NONInt_th_completeness_CheckV_HMM-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_completeness_CheckV_HMM" title="Permalink"></a></h4><ul><li>this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is HMM</li><li>value is Float</li><li>default is 10.0</li></ul><h3 id="NONInt_th_num_predictors_CheckV_DTR_ITR_AAI"><a class="docs-heading-anchor" href="#NONInt_th_num_predictors_CheckV_DTR_ITR_AAI"><code>NONInt_th_num_predictors_CheckV_DTR_ITR_AAI</code></a><a id="NONInt_th_num_predictors_CheckV_DTR_ITR_AAI-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_num_predictors_CheckV_DTR_ITR_AAI" title="Permalink"></a></h3><ul><li>this parameters gives the minimum number of initial predictions necessary to keep a viral contigs if its completeness was determined based on the presence of direct-terminal repeats (DTRs) or inverted-terminal repeats (ITRs) combined with AAI (see CheckV documentation)</li><li>value is Integer</li><li>default is 1</li></ul><h3 id="NONInt_th_num_predictors_CheckV_DTR_ITR_HMM"><a class="docs-heading-anchor" href="#NONInt_th_num_predictors_CheckV_DTR_ITR_HMM"><code>NONInt_th_num_predictors_CheckV_DTR_ITR_HMM</code></a><a id="NONInt_th_num_predictors_CheckV_DTR_ITR_HMM-1"></a><a class="docs-heading-anchor-permalink" href="#NONInt_th_num_predictors_CheckV_DTR_ITR_HMM" title="Permalink"></a></h3><ul><li>this parameters gives the minimum number of initial predictions necessary to keep a viral contigs if its completeness was determined based on the presence of direct-terminal repeats (DTRs) or inverted-terminal repeats (ITRs) combined with HMM (see CheckV documentation)</li><li>value is Integer</li><li>default is 1</li></ul><h2 id="INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs"><a class="docs-heading-anchor" href="#INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs">INTEGRATED viruses branch: arguments to select final viral contigs</a><a id="INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs-1"></a><a class="docs-heading-anchor-permalink" href="#INTEGRATED-viruses-branch:-arguments-to-select-final-viral-contigs" title="Permalink"></a></h2><p>At the end of the workflow DoViP keeps only the INTEGRATED viruses that fulfill certain criteria. The selection depends on the method used by CheckV to estimate the viral contig completeness, on the value of the viral contig completeness, and on the initial predictor coverage value (average number of predictors at each base along the viral sequence) for the respective viral sequence.   </p><h3 id="If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:-2"><a class="docs-heading-anchor" href="#If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:-2">If CheckV could not estimate the completeness of the viral contigs:</a><a class="docs-heading-anchor-permalink" href="#If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:-2" title="Permalink"></a></h3><ul><li>it means that CheckV could not find similarities in its database</li><li>than only the number of viral predictors is taken into consideration. </li><li>I recommend using a higher number of initial predictors, to increase the confidence that the respective contig is actually viral  </li></ul><h4 id="Int_th_num_predictors_CheckV_NA"><a class="docs-heading-anchor" href="#Int_th_num_predictors_CheckV_NA"><code>Int_th_num_predictors_CheckV_NA</code></a><a id="Int_th_num_predictors_CheckV_NA-1"></a><a class="docs-heading-anchor-permalink" href="#Int_th_num_predictors_CheckV_NA" title="Permalink"></a></h4><ul><li>this parameter gives the minimum number of initial predictors necessary to keep a viral contig if completeness is NA</li><li>value is Integer</li><li>default is 2.5</li></ul><h3 id="If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:-2"><a class="docs-heading-anchor" href="#If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:-2">If CheckV used AAIHighConf method to estimate the completeness of the viral contigs:</a><a class="docs-heading-anchor-permalink" href="#If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:-2" title="Permalink"></a></h3><ul><li>it means that it found related viruses in its database and it can estimate the completeness with high precision</li><li>all viral contigs with a completeness of at least 90% are kept, regardless of the number of initial predictors </li></ul><h4 id="Int_th_num_predictors_CheckV_AAIHighConf"><a class="docs-heading-anchor" href="#Int_th_num_predictors_CheckV_AAIHighConf"><code>Int_th_num_predictors_CheckV_AAIHighConf</code></a><a id="Int_th_num_predictors_CheckV_AAIHighConf-1"></a><a class="docs-heading-anchor-permalink" href="#Int_th_num_predictors_CheckV_AAIHighConf" title="Permalink"></a></h4><ul><li>this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is AAIHighConf</li><li>value is Integer</li><li>default is 1</li></ul><h4 id="Int_th_completeness_CheckV_AAIHighConf"><a class="docs-heading-anchor" href="#Int_th_completeness_CheckV_AAIHighConf"><code>Int_th_completeness_CheckV_AAIHighConf</code></a><a id="Int_th_completeness_CheckV_AAIHighConf-1"></a><a class="docs-heading-anchor-permalink" href="#Int_th_completeness_CheckV_AAIHighConf" title="Permalink"></a></h4><ul><li>this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is AAIHighConf</li><li>value is Float</li><li>default is 30.0</li></ul><h3 id="If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:-2"><a class="docs-heading-anchor" href="#If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:-2">If CheckV used AAIMediumConf method to estimate the completeness of the viral contigs:</a><a class="docs-heading-anchor-permalink" href="#If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:-2" title="Permalink"></a></h3><ul><li>it means that it found related viruses in its database and it can estimate the completeness with moderate precission</li></ul><h4 id="Int_th_num_predictors_CheckV_AAIMediumConf"><a class="docs-heading-anchor" href="#Int_th_num_predictors_CheckV_AAIMediumConf"><code>Int_th_num_predictors_CheckV_AAIMediumConf</code></a><a id="Int_th_num_predictors_CheckV_AAIMediumConf-1"></a><a class="docs-heading-anchor-permalink" href="#Int_th_num_predictors_CheckV_AAIMediumConf" title="Permalink"></a></h4><ul><li>this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is AAIMediumConf</li><li>value is Integer</li><li>default is 1.5    </li></ul><h4 id="Int_th_completeness_CheckV_AAIMediumConf"><a class="docs-heading-anchor" href="#Int_th_completeness_CheckV_AAIMediumConf"><code>Int_th_completeness_CheckV_AAIMediumConf</code></a><a id="Int_th_completeness_CheckV_AAIMediumConf-1"></a><a class="docs-heading-anchor-permalink" href="#Int_th_completeness_CheckV_AAIMediumConf" title="Permalink"></a></h4><ul><li>this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is AAIMediumConf</li><li>value is Float</li><li>default is 10.0</li></ul><h3 id="If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:-2"><a class="docs-heading-anchor" href="#If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:-2">If CheckV used HMM method to estimate the completeness of the viral contigs:</a><a class="docs-heading-anchor-permalink" href="#If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:-2" title="Permalink"></a></h3><ul><li>it means that it found distantly related viruses in its database and it can estimate the completeness with low precision</li></ul><h4 id="Int_th_num_predictors_CheckV_HMM"><a class="docs-heading-anchor" href="#Int_th_num_predictors_CheckV_HMM"><code>Int_th_num_predictors_CheckV_HMM</code></a><a id="Int_th_num_predictors_CheckV_HMM-1"></a><a class="docs-heading-anchor-permalink" href="#Int_th_num_predictors_CheckV_HMM" title="Permalink"></a></h4><ul><li>this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is HMM</li><li>value is Integer</li><li>default is 1.5        </li></ul><h4 id="Int_th_completeness_CheckV_HMM"><a class="docs-heading-anchor" href="#Int_th_completeness_CheckV_HMM"><code>Int_th_completeness_CheckV_HMM</code></a><a id="Int_th_completeness_CheckV_HMM-1"></a><a class="docs-heading-anchor-permalink" href="#Int_th_completeness_CheckV_HMM" title="Permalink"></a></h4><ul><li>this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is HMM</li><li>value is Float</li><li>default is 10.0</li></ul></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../Running/">« Overview</a><a class="docs-footer-nextpage" href="../Running_multipleworkflow_params/">Multiple workflows parameters »</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.10.2 on <span class="colophon-date" title="Tuesday 6 May 2025 06:30">Tuesday 6 May 2025</span>. Using Julia version 1.10.0.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
