<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Outputs - step by step breakdown · DoViP</title><meta name="title" content="Outputs - step by step breakdown · DoViP"/><meta property="og:title" content="Outputs - step by step breakdown · DoViP"/><meta property="twitter:title" content="Outputs - step by step breakdown · DoViP"/><meta name="description" content="Documentation for DoViP."/><meta property="og:description" content="Documentation for DoViP."/><meta property="twitter:description" content="Documentation for DoViP."/><script data-outdated-warner src="../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL=".."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../assets/documenter.js"></script><script src="../search_index.js"></script><script src="../siteinfo.js"></script><script src="../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../">DoViP</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li><a class="tocitem" href="../">Home</a></li><li><span class="tocitem">Installation</span><ul><li><a class="tocitem" href="../Installation/">Overview</a></li><li><a class="tocitem" href="../installation_conda_dependencies/">Dependencies</a></li><li><a class="tocitem" href="../installation_DoViP_env/">DoViP environment</a></li></ul></li><li><span class="tocitem">Running</span><ul><li><a class="tocitem" href="../Running/">Overview</a></li><li><a class="tocitem" href="../Running_singleworkflow_params/">Single workflow parameters</a></li><li><a class="tocitem" href="../Running_multipleworkflow_params/">Multiple workflows parameters</a></li></ul></li><li><span class="tocitem">DoViP projects and outputs</span><ul><li><a class="tocitem" href="../dovip_projects/">Project definition</a></li><li><a class="tocitem" href="../dovip_outputs_overview/">Outputs - overview</a></li><li class="is-active"><a class="tocitem" href>Outputs - step by step breakdown</a><ul class="internal"><li><a class="tocitem" href="#Removal-of-small-contigs"><span>Removal of small contigs</span></a></li><li><a class="tocitem" href="#Running-initial-predictors"><span>Running initial predictors</span></a></li><li><a class="tocitem" href="#Merging-and-checkV-for-non-integrated-viruses"><span>Merging and checkV for non-integrated viruses</span></a></li><li><a class="tocitem" href="#Merging-and-checkV-for-integrated-viruses"><span>Merging and checkV for integrated viruses</span></a></li><li><a class="tocitem" href="#unMixing-viruses"><span>unMixing viruses</span></a></li><li><a class="tocitem" href="#Predicting-virus-lifestyles-with-PhaTYP"><span>Predicting virus lifestyles with PhaTYP</span></a></li><li><a class="tocitem" href="#Taxonomic-assignment-of-non-integrated-viral-sequences"><span>Taxonomic assignment of non-integrated viral sequences</span></a></li><li><a class="tocitem" href="#Final-virus-selection-non-integrated-virus-branch"><span>Final virus selection - non-integrated virus branch</span></a></li><li><a class="tocitem" href="#Taxonomic-assignment-of-integrated-viral-sequences"><span>Taxonomic assignment of integrated viral sequences</span></a></li><li><a class="tocitem" href="#Final-virus-selection-integrated-virus-branch"><span>Final virus selection - integrated virus branch</span></a></li></ul></li><li><a class="tocitem" href="../continue_projs/">Continue projects</a></li></ul></li><li><a class="tocitem" href="../releases/">Release notes</a></li><li><a class="tocitem" href="../authors/">Authors</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li><a class="is-disabled">DoViP projects and outputs</a></li><li class="is-active"><a href>Outputs - step by step breakdown</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Outputs - step by step breakdown</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/CristinaMoraru/DoViP" title="View the repository on GitHub"><span class="docs-icon fa-brands"></span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/CristinaMoraru/DoViP/blob/main/docs_make/src/dovip_outputs_stepbystep.md" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="Outputs-step-by-step-breakdown"><a class="docs-heading-anchor" href="#Outputs-step-by-step-breakdown">Outputs - step by step breakdown</a><a id="Outputs-step-by-step-breakdown-1"></a><a class="docs-heading-anchor-permalink" href="#Outputs-step-by-step-breakdown" title="Permalink"></a></h1><h2 id="Removal-of-small-contigs"><a class="docs-heading-anchor" href="#Removal-of-small-contigs">Removal of small contigs</a><a id="Removal-of-small-contigs-1"></a><a class="docs-heading-anchor-permalink" href="#Removal-of-small-contigs" title="Permalink"></a></h2><p>Initialy, DoViP removes from the input file all contigs smaller than <code>min_contig_length</code>. The resulting fasta file is saved a folder with the sufix coresponding to <code>min_contig_length</code>, as shown below. </p><pre><code class="language-plaintext hljs">single_workflow_project_name/ 
    ├──  00_ALL_min_length_#   (#=`min_contig_length`)
    |        ├── outMFasta_small.fna  
    ├──  ... </code></pre><h2 id="Running-initial-predictors"><a class="docs-heading-anchor" href="#Running-initial-predictors">Running initial predictors</a><a id="Running-initial-predictors-1"></a><a class="docs-heading-anchor-permalink" href="#Running-initial-predictors" title="Permalink"></a></h2><p>The .fna file from the removal of small contigs step is sent to each of the initial predictions selected by the user. The results from each intial predictor are saved into folders with corresponding names, within subfolders labeled with the sufix <code>_out</code>. The &quot;raw&quot; data from each predictor are further postprocessed by DoViP and saved saved in two .tsv files, one for the integrated viruses branch (geNomad, VirSorter2 and VIBRANT), and the other for the non-integrated viruses branch (geNomad, DeepVir Finder, VirSorter2, VIBRANT and ViralVerify). The two .tsv files are saved directly in these folders, as shown below.  </p><pre><code class="language-plaintext hljs">single_workflow_project_name/ 
    ├──  ... 
    ├──  01a_ALL_genomad
    |        ├── 01a_ALL_01_genomad_out
    |        ├── 01a_ALL_02_postgenomad_integrated_df.tsv
    |        ├── 01a_ALL_02_postgenomad_nonintegrated_df.tsv
    ├── 01b_ALL_DVF
    |        ├── 01b_ALL_01_DVF_out
    |        ├── 01b_ALL_postdfvf_nonintegrated_df.tsv
    ├── 01c_ALL_virSorter2
    |        ├── 01c_ALL_01_virSorter2_out
    |        ├── 01c_ALL_02_postVS2_integrated_df.tsv
    |        ├── 01c_ALL_02_postVS2_nonintegrated_df.tsv
    ├── 01d_ALL_vibrant
    |        ├── 01d_ALL_01_vibrant_out
    |        ├── 01d_ALL_02_postvibrant_integrated_df.tsv
    |        ├── 01d_ALL_02_postvibrant_nonintegrated_df.tsv
    ├── 01e_ALL_viralVerify
    |        ├── 01e_ALL_01_viralVerify_out
    |        ├── 01e_ALL_02_postViralVerify.tsv 
    ├──  ... </code></pre><h2 id="Merging-and-checkV-for-non-integrated-viruses"><a class="docs-heading-anchor" href="#Merging-and-checkV-for-non-integrated-viruses">Merging and checkV for non-integrated viruses</a><a id="Merging-and-checkV-for-non-integrated-viruses-1"></a><a class="docs-heading-anchor-permalink" href="#Merging-and-checkV-for-non-integrated-viruses" title="Permalink"></a></h2><p>In this step, DoViP first brings toghether all virus contigs estimated as lytic by the various predictors and their associated information, and saves them in a .tsv and a .fna file with the prefix <code>02_N-01_00</code>. The .fna file then is inputed in CheckV, and the results are saved in the <code>02_N-01_01_CheckV</code> folder. Postprocessing after checkV includes saving the contigs marked as containing integrated viruses (they will be inputed in the integrated viruses branch) and then saving the remaining non-integrated viruses and their coresponding information in a .tsv file (prefix <code>02_N-01_02</code>). The folder structure and outputs are shown below.  </p><pre><code class="language-plaintext hljs">single_workflow_project_name/ 
    ├── ...       
    ├── 02_N-01_checkV_Nonintegrated
    |        ├── 02_N-01_00_aggregated_nonintegrated_virus_contigs.tsv
    |        ├── 02_N-01_00_aggregated_nonintegrated_virus_contigs.fna
    |        ├── 02_N-01_01_checkV
    |        ├── 02_N-01_02_postcheckV_nonintegrated_df.tsv
    |        ├── 02_N-01_02_postcheckV_integrated.tsv.fna
    ├──  ... </code></pre><h2 id="Merging-and-checkV-for-integrated-viruses"><a class="docs-heading-anchor" href="#Merging-and-checkV-for-integrated-viruses">Merging and checkV for integrated viruses</a><a id="Merging-and-checkV-for-integrated-viruses-1"></a><a class="docs-heading-anchor-permalink" href="#Merging-and-checkV-for-integrated-viruses" title="Permalink"></a></h2><p>In this step DoViP does the following:</p><ul><li>brings together all virus contigs estimated as &quot;temperate&quot; by the various predictors (including those from the 02<em>N-01 step, predicted by CheckV as integrated), and saves them in a .fna and a .tsv file with the prefix `03</em>I-01_00`.</li><li>inputs the .fna file in CheckV, and saves the results in the <code>03_I-01_01_checkVIntegrated_1</code> folder.</li><li>uses the information from the initial predictors and CheckV to remove host regions and to merge overlapping viral regions, and saved the them in a .fna and .tsv file with the prefix <code>03_I-01_03</code> (the filex with I<em>02</em>02 represent intermediary results).</li><li>sends the merged integrated viruses to CheckV, and saves the data in the folder <code>03_I-01_04_checkVIntegrated_2</code></li><li>postprocesses the output from the second CheckV step to confirm that no host-contamination is present and to add the completeness and other checkV data to the merged integrated viruses. The data are saved in the file with the <code>03_I-01_05</code> prefix.</li></ul><pre><code class="language-plaintext hljs">single_workflow_project_name/ 
    ├── ...
    ├── 03_I-01_checkV_Integrated
    |        ├── 03_I-01_00_aggregated_integrated_virus_All_contigs.tsv
    |        ├── 03_I-01_00_aggregated_integrated_virus_All_contigs.fna
    |        ├── 03_I-01_01_checkVIntegrated_1
    |        ├── 03_I-01_02_postcheckV1_integrated_cordf.tsv
    |        ├── 03_I-01_02_postcheckV1_integrated_withmergedprovirIDs_cordf.tsv
    |        ├── 03_I-01_03_merged_integrated.tsv
    |        ├── 03_I-01_03_merged_integrated.fna
    |        ├── 03_I-01_04_checkVIntegrated_2
    |        ├── 03_I-01_05_postcheckV2_integrated_df_p.tsv
    ├──  ... </code></pre><h2 id="unMixing-viruses"><a class="docs-heading-anchor" href="#unMixing-viruses">unMixing viruses</a><a id="unMixing-viruses-1"></a><a class="docs-heading-anchor-permalink" href="#unMixing-viruses" title="Permalink"></a></h2><p>It can happen that the same contig is identified by some initial predictors as non-integrated viruses and by some other predictors as harboring integrated viruses. In the &quot;unMixing viruses&quot; step, DoViP does the following:</p><ul><li>it uses <code>02_N-01_02_postcheckV_nonintegrated_df.tsv</code> and <code>03_I-01_05_postcheckV2_integrated_df_p.tsv</code> files to identify these mixed predictions, and reports them in the <code>M_02_Detection</code> folder (see files with the prefix <code>04_M_00</code>).   </li><li>it considers all &quot;mixed&quot; contigs coming from the non-integrated as truly non-integrated, because CheckV found no host contamination present. Furtheremore, DoViP considers all &quot;mixed&quot; viral sequences comming from the integrated branch as beeing incorectly labeled as integrated. Therefore, for the &quot;mixed&quot; contigs, DoViP merges the integrated and non-integrated predictions by transfering the intial predictors from the integrated branch to the nonintegrated virus contigs. This will increase the number of initial predictors for a partticular contig, increasing therefore its chance to pass the thresholds from the final step. The new information on the non-integrated viruses will be saved in  <code>04_M_01_resolved_NonInt_Viruses.tsv</code>, <code>04_M_02_resolved_nonintegrated.fna</code> and <code>04_M_02_resolved_nonintegrated_trimmed_DTR.fna</code>.   </li><li>it removes all &quot;mixed&quot; viral contigs from the <code>03_I-01_05_postcheckV2_integrated_df_p.tsv</code>, and the updated integrated viruses are saved in <code>04_M_01_resolved_Int_Viruses.tsv</code> and <code>04_M_02_resolved_integrated.fna</code>.      </li></ul><pre><code class="language-plaintext hljs">single_workflow_project_name/ 
    ├── ...
    ├── 04_M_Detection
    |        ├── 04_M_00_Mixed_Int_Viruses.tsv 
    |        ├── 04_M_00_Mixed_NonInt_Viruses.tsv
    |        ├── 04_M_01_resolved_NonInt_Viruses.tsv
    |        ├── 04_M_01_resolved_Int_Viruses.tsv
    |        ├── 04_M_02_resolved_nonintegrated.fna
    |        ├── 04_M_02_resolved_nonintegrated_trimmed_DTR.fna
    |        ├── 04_M_02_resolved_integrated.fna
    └── ...</code></pre><h2 id="Predicting-virus-lifestyles-with-PhaTYP"><a class="docs-heading-anchor" href="#Predicting-virus-lifestyles-with-PhaTYP">Predicting virus lifestyles with PhaTYP</a><a id="Predicting-virus-lifestyles-with-PhaTYP-1"></a><a class="docs-heading-anchor-permalink" href="#Predicting-virus-lifestyles-with-PhaTYP" title="Permalink"></a></h2><p>In this step DoViP sends the non-integrated viruses from the previous step (<code>04_M_02_resolved_nonintegrated.fna</code>) to PhaTYP, which predicts their potential life styles. The PhaTYP results are saved in the folder with the <code>_out</code> suffix. The predictions are then added to the previous data and saved in the .tsv file with the <code>05_N-02_02</code> prefix.</p><pre><code class="language-plaintext hljs">single_workflow_project_name/ 
    ├── ...
    ├── 05_N-02_phaTYPNonintegrated
    |        ├── 05_N-02_01_phaTYP_out
    |        ├── 05_N-02_02_NonINT_mergedPostCheckV_PhaTYP.tsv</code></pre><h2 id="Taxonomic-assignment-of-non-integrated-viral-sequences"><a class="docs-heading-anchor" href="#Taxonomic-assignment-of-non-integrated-viral-sequences">Taxonomic assignment of non-integrated viral sequences</a><a id="Taxonomic-assignment-of-non-integrated-viral-sequences-1"></a><a class="docs-heading-anchor-permalink" href="#Taxonomic-assignment-of-non-integrated-viral-sequences" title="Permalink"></a></h2><p>Here DoviP sends the non-integrated viruses from the unMixing viruses step (<code>04_M_02_resolved_nonintegrated.fna</code>) to geNomad, to predicting their taxonomy (including for the viral sequences predicted by other initial predictors than geNomad). The taxonomic predictions are added then to the output table from the PhaTYP step (<code>05_N-02_02_NonINT_mergedPostCheckV_PhaTYP.tsv</code>).  </p><pre><code class="nohighlight hljs">    ├── 06_N-03_geNomad_taxonomy_all_non-integrated_viral_sequences
    |        ├── 06_N-03_02_postgenomad_tax.tsv    </code></pre><h2 id="Final-virus-selection-non-integrated-virus-branch"><a class="docs-heading-anchor" href="#Final-virus-selection-non-integrated-virus-branch">Final virus selection - non-integrated virus branch</a><a id="Final-virus-selection-non-integrated-virus-branch-1"></a><a class="docs-heading-anchor-permalink" href="#Final-virus-selection-non-integrated-virus-branch" title="Permalink"></a></h2><p>DoViP uses data in the <code>06_N-03_02_postgenomad_tax.tsv</code> file to perform a final selection of the non-integrated virus contigs based on their completeness and the number of initial predictors. The results are saved as .fna and .tsv file, as shown below. </p><pre><code class="language-plaintext hljs">single_workflow_project_name/ 
    ├── ...
    ├── 07_N-04_FinalThresholding
    |        ├── 07_N-04_Final_Viruses_df.tsv
    |        ├── 07_N-04_Final_Viruses.fna
    |        ├── 07_N-04_Final_Viruses_trimmed_DTR.fna
    └── ...</code></pre><h2 id="Taxonomic-assignment-of-integrated-viral-sequences"><a class="docs-heading-anchor" href="#Taxonomic-assignment-of-integrated-viral-sequences">Taxonomic assignment of integrated viral sequences</a><a id="Taxonomic-assignment-of-integrated-viral-sequences-1"></a><a class="docs-heading-anchor-permalink" href="#Taxonomic-assignment-of-integrated-viral-sequences" title="Permalink"></a></h2><p>Here DoviP sends the integrated viruses from the unMixing viruses step (<code>04_M_02_resolved_integrated.fna</code>) to geNomad, to predicting their taxonomy (including for the viral sequences predicted by other initial predictors than geNomad).  The taxonomic predictions are added then to the output table from the unMixing viruses step (<code>04_M_01_resolved_Int_Viruses.tsv</code>).  </p><pre><code class="nohighlight hljs">    ├── 08_I-02_geNomad_taxonomy_all_integrated_viral_sequences
    |        ├──  08_I-02_02_postgenomad_tax.tsv        </code></pre><h2 id="Final-virus-selection-integrated-virus-branch"><a class="docs-heading-anchor" href="#Final-virus-selection-integrated-virus-branch">Final virus selection - integrated virus branch</a><a id="Final-virus-selection-integrated-virus-branch-1"></a><a class="docs-heading-anchor-permalink" href="#Final-virus-selection-integrated-virus-branch" title="Permalink"></a></h2><p>DoViP uses data in the <code>08_I-02_02_postgenomad_tax.tsv</code> file to perform a final selection of the integrated virus contigs based on their completeness and the number of initial predictors. The results are saved as .fna and .tsv file, as shown below. </p><pre><code class="language-plaintext hljs">single_workflow_project_name/ 
    ├── ...
    ├── 09_I-01_FinalThresholding
    |        ├── 09_I-03_Final_Viruses_df.tsv
    |        ├── 09_I-03_Final_Viruses.fna
    └── ...</code></pre></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../dovip_outputs_overview/">« Outputs - overview</a><a class="docs-footer-nextpage" href="../continue_projs/">Continue projects »</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.10.2 on <span class="colophon-date" title="Saturday 3 May 2025 12:56">Saturday 3 May 2025</span>. Using Julia version 1.10.0.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
