var documenterSearchIndex = {"docs":
[{"location":"installation_conda_dependencies/#DoViP-dependencies-and-their-installation","page":"Dependencies","title":"DoViP dependencies and their installation","text":"","category":"section"},{"location":"installation_conda_dependencies/#geNomad","page":"Dependencies","title":"geNomad","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"geNomad use and installation is described in detail here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"mamba create -p path/to/conda_genomad -c conda-forge -c bioconda genomad\n\n# database setup\nconda activate conda_genomad\ncd /path/for/databases\ngenomad download-database .","category":"page"},{"location":"installation_conda_dependencies/#VIBRANT","page":"Dependencies","title":"VIBRANT","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"VIBRANT use and installation is described in detail here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# env\nconda create -n conda_VIBRANT\nconda activate conda_VIBRANT\n\n# dependencies\nconda install -c bioconda hmmer\nconda install -c bioconda prodigal\nconda install -c bioconda biopython\nconda install -c anaconda pandas\nconda install matplotlib\nconda install seaborn\nconda install -c anaconda numpy\nconda install scikit-learn\n\n# install VIBRANT itself\ncd path/for/vibrant/repo\ngit clone https://github.com/AnantharamanLab/VIBRANT\nchmod -R 777 VIBRANT/\n\n# database\ncd databases\npip install --upgrade scikit-learn==0.21.3\npip install --upgrade numpy==1.17.0\npython3 VIBRANT_setup.py ","category":"page"},{"location":"installation_conda_dependencies/#VirSorter2","page":"Dependencies","title":"VirSorter2","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"VirSorter 2 use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# create env and instal, virsorter2\nconda create -p path/to/conda_virsorter2\nmamba activate path/to/conda_virsorter2\nmamba install -c conda-forge -c bioconda virsorter=2\n\n# databases\ncd conda_virsorter2/\nvirsorter setup -d db -j 4","category":"page"},{"location":"installation_conda_dependencies/#DeepVir-Finder","page":"Dependencies","title":"DeepVir Finder","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"DeepVir Finder use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# create env\nconda create -p /path/to/conda_DVF python=3.6 numpy theano=1.0.3 keras=2.2.4 scikit-learn Biopython h5py\nconda activate conda_DVF\n\n# clone the git repo\ncd path/for/repo\ngit clone https://github.com/jessieren/DeepVirFinder","category":"page"},{"location":"installation_conda_dependencies/#CheckV","page":"Dependencies","title":"CheckV","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"CheckV use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# env\nconda create -p path/to/conda_checkV\nconda activate conda_checkV\n\n#\nconda install -c conda-forge -c bioconda checkV\n\n\n# database\ncd path/for/databases/\ncheckv download_database ./","category":"page"},{"location":"installation_conda_dependencies/#PhaBox","page":"Dependencies","title":"PhaBox","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"PhaBox use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"cd path/for/repo\ngit clone https://github.com/KennthShang/PhaBOX\ncd PhaBOX/\nconda env create -f webserver.yml -p path/for/conda_PhaBOX\nconda activate conda_PhaBOX\n\n# databases\npip install gdown\ngdown  --id 1hjACPsIOqqcS5emGaduYvYrCzrIpt2_9\ngdown  --id 1E94ii3Q0O8ZBm7UsyDT_n06YekNtfV20\n\npath/for/repo/unzip phagesuite_database.zip  > /dev/null\npath/for/repo/unzip phagesuite_parameters.zip  > /dev/null\n\n\ncp blastxml_to_tabular.pypath/for/conda_PhaBOX/bin/blastxml_to_tabular.py\nchmod 777 path/for/conda_PhaBOX/bin/blastxml_to_tabular.py","category":"page"},{"location":"installation_DoViP_env/#Variant-1:-Step-by-step","page":"DoViP environment","title":"Variant 1: Step by step","text":"","category":"section"},{"location":"installation_DoViP_env/#1.-Create-the-conda-environment-for-DoViP-and-install-Julia-programming-language","page":"DoViP environment","title":"1. Create the conda environment for DoViP and install Julia programming language","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"# create conda env\nconda create -p /path/toward/conda_DoViP\nconda activate /path/toward/conda_DoViP\n\n# download and install julia version 1.10.0\ncd /path/toward/conda_DoViP\nwget https://julialang-s3.julialang.org/bin/linux/x64/1.10/julia-1.10.0-linux-x86_64.tar.gz\ntar -xzvf julia-1.10.0-linux-x86_64.tar.gz\nrm julia-1.10.0-linux-x86_64.tar.gz\nmkdir bin\n\n# make a symbolic link of the julia executable in the /path/toward/conda_DoViP/bin/ folder\nln -s /path/toward/conda_DoViP/julia-1.10.0/bin/julia /path/toward/conda_DoViP/bin/julia\n","category":"page"},{"location":"installation_DoViP_env/#2.-Bring-the-DoViP-repo","page":"DoViP environment","title":"2. Bring the DoViP repo","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"cd /path/toward/conda_DoViP\ngit clone https://github.com/CristinaMoraru/DoViP.git\n\nmv DoViP/* .\nrm DoViP\n\n# give permissions to the folder where julia saved its libraries\ncd share\nchmod 777 -R julia","category":"page"},{"location":"installation_DoViP_env/#3.-Refresh-the-DoViP-conda-environment","page":"DoViP environment","title":"3. Refresh the DoViP conda environment","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"conda deactivate conda_DoViP\nconda activate /path/toward/conda_DoViP\n","category":"page"},{"location":"installation_DoViP_env/#4.-Instantiate-DoViP-libraries","page":"DoViP environment","title":"4. Instantiate DoViP libraries","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"\ncd /path/toward/conda_DoViP/\n\njulia -e 'using Pkg; Pkg.activate(\"DoViP_App.jl\"); Pkg.instantiate()' \n","category":"page"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"Instantiation is necessary to bring on your machine the packages on which DoViP depends. After instantiation, it will attempt to start DoViP already, and therefore you will see an error message:  \"\". Ignore it.","category":"page"},{"location":"installation_DoViP_env/#5.Test-installation","page":"DoViP environment","title":"5.Test installation","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"DoViP --help","category":"page"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"You should get the following message: \"DoViP - a workflow for virus prediction in metagenomes.\"","category":"page"},{"location":"installation_DoViP_env/#Variant-2:-From-a-conda-package","page":"DoViP environment","title":"Variant 2: From a conda package","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"to be prepared","category":"page"},{"location":"Running_singleworkflow_params/#DoViP-singleworklow-parameters","page":"Single workflow parameters","title":"DoViP singleworklow parameters","text":"","category":"section"},{"location":"Running_singleworkflow_params/#General-project-options","page":"Single workflow parameters","title":"General project options","text":"","category":"section"},{"location":"Running_singleworkflow_params/#inref","page":"Single workflow parameters","title":"inref","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the fasta file containing the input contigs. \nAccepted file extensions are: \".fasta\", \".fna\", \".fst\", \".fas\", \".fa\", \".mfa\"\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#projtype","page":"Single workflow parameters","title":"projtype","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the only acceptable value here is 'singleworkflow', and it tells DoViP that only one metagenome is analysed for this row.\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#sample_set","page":"Single workflow parameters","title":"sample_set","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the name of the sample/metagenome collection\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#use_slurm","page":"Single workflow parameters","title":"use_slurm","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"if calculations should be run on an High Performance Computing Cluster, using slurm and sbatch\npossible values: 'true' or 'false'\ndefault is 'false'","category":"page"},{"location":"Running_singleworkflow_params/#continue","page":"Single workflow parameters","title":"continue","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"if the calculations for this sample should be continued from a previous run of the project\nthis is useful for example when changing the desired initial predictors (see details below) or the thresholds for the final selection of the viral contigs\npossible values: 'true' or 'false'\ndefault is 'false'","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"!!! Note:      if continue is set to 'false' and the project folder allready exists, a message will appear asking to confirm or deny the deletion of the existing folder.","category":"page"},{"location":"Running_singleworkflow_params/#stop_after_initial_predictors","page":"Single workflow parameters","title":"stop_after_initial_predictors","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"if the calculations should stop after the initial predictions have finished (useful for example if the project should be continued later)\npossible values: 'true' or 'false'\ndefault is 'false'","category":"page"},{"location":"Running_singleworkflow_params/#min_contig_length","page":"Single workflow parameters","title":"min_contig_length","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the minimum length of the contigs from the input file (inref) which should be kept\nonly contigs equal or larger than this value will be sent to the intial predictors\nvalue is Integer\ndefault value is 1000","category":"page"},{"location":"Running_singleworkflow_params/#PARAMETERS-FOR-INITIAL-PREDICTION-STEPS","page":"Single workflow parameters","title":"PARAMETERS FOR INITIAL PREDICTION STEPS","text":"","category":"section"},{"location":"Running_singleworkflow_params/#geNomad-options","page":"Single workflow parameters","title":"geNomad options","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"If this this step is selected for running (genomad_signal=do), the following will happen:  ","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"DoViP will start a geNomad run in the conda environment given by the genomad_env parameter, using the following command:\n`genomad end-to-end --cleanup --splits 8 '/path/to/input.fasta' '/path/to/output_folder' `genomadDB_p` --min-score `genomad_min_score``\nDoViP will postprocess the geNomad results (file _virus_summary.tsv) to separate NON-INTEGRATED from INTEGRATED viruses.","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"If, instead of running geNomad, external results should be used, the following will happen:","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"DoViP will postprocess the geNomad results (file _virus_summary.tsv) to separate NON-INTEGRATED from INTEGRATED viruses.","category":"page"},{"location":"Running_singleworkflow_params/#genomad_signal","page":"Single workflow parameters","title":"genomad_signal","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"if genomad predictions should be considered\npossible values:\n'do' - run this predictor step. Can be always used, regardless of the signals in the previous steps\n'dont' - don't run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a 'dont' or 'remove' signal in the previous step.\n'use'- re-use predictor results calculated in previous runs of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished.\n'use_external' - use results calculated previously by the predictor independently from a DoViP project.\n'ignore' - doesn't take into consideration existing results calculated in a previous run of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished. \n'remove' - removes results calculated in a previous step. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or ignore in the previous step, regardless of their \"progress\". \nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#genomad_res","page":"Single workflow parameters","title":"genomad_res","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with geNomad results that were calculated outside DoViP\nneeded only if genomad_signal=use_external","category":"page"},{"location":"Running_singleworkflow_params/#genomad_env","page":"Single workflow parameters","title":"genomad_env","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the conda environment where geNomad is installed\nmandatory parameter if geNomad signal is do, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#genomadDB_p","page":"Single workflow parameters","title":"genomadDB_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with the geNomad database\nmandatory parameter if geNomad signal is do, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#genomad_min_score","page":"Single workflow parameters","title":"genomad_min_score","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"mininum score to be used by geNomad to filter viral contigs\nvalue is Float (between 0.0 and 1.0)\ndefault it 0.7","category":"page"},{"location":"Running_singleworkflow_params/#genomad_sbatch_time","page":"Single workflow parameters","title":"genomad_sbatch_time","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"maximum time that a sbatch job running geNomad (in slurm) is allowed to run\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#genomad_cpus_per_task","page":"Single workflow parameters","title":"genomad_cpus_per_task","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum number of CPUs in which geNomad should be run (either on a local server or on HPC).\nwhen running on an HPC, this parameter will be further transmited to sbatch\nvalue is Integer\ndefault is 20 ","category":"page"},{"location":"Running_singleworkflow_params/#genomad_sbatch_mem","page":"Single workflow parameters","title":"genomad_sbatch_mem","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum memmory requirements if geNomad is run as an sbatch job\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given ","category":"page"},{"location":"Running_singleworkflow_params/#DeepVir-Finder-options","page":"Single workflow parameters","title":"DeepVir Finder options","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"If this this step is selected for running (DVF_signal=do), the following will happen:   ","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"DoViP will check each contig in the input file, and, if larger than the value of DVF_maxContigLen it will cut it in smaller pieces. This is to avoid DVF freezing when it encounters larger than 2099000.\nDoViP will start a DVF run in the conda environment given by parameter DVF_env with the following command:  \npython 'path/to/dvf.py' -i 'path/to/input.fasta' -o' path/to/output/folder' -l 'min_contig_len' -c `DVF_cpus_per_task`\nDoViP will merge back into a single entry the larger contigs which have been split in the first step, with the score and the p-value repersentig the median of the smaller contigs.\nDoViP will postprocess all DVF results to remove all contigs with a score smaller than DVF_scoreTh and a p-value larger than DVF_pThreshold. Only NON-INTEGRATED viruses are outputed by DVF.","category":"page"},{"location":"Running_singleworkflow_params/#DVF_signal","page":"Single workflow parameters","title":"DVF_signal","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"if DeepVir Finder predictions should be considered\npossible values:\n'do' - run this predictor step. Can be always used, regardless of the signals in the previous steps\n'dont' - don't run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a 'dont' or 'remove' signal in the previous step.\n'use'- re-use predictor results calculated in previous runs of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished.\n'use_external' - use results calculated previously by the predictor independently from a DoViP project.\n'ignore' - doesn't take into consideration existing results calculated in a previous run of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished. \n'remove' - removes results calculated in a previous step. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or ignore in the previous step, regardless of their \"progress\". \nmandatory parameter if DVF should be run, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#DVF_env","page":"Single workflow parameters","title":"DVF_env","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the conda environment where DVF is installed\nmandatory parameter, no default is given\t","category":"page"},{"location":"Running_singleworkflow_params/#DVF_maxContigLen","page":"Single workflow parameters","title":"DVF_maxContigLen","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum length that a contig should have for DVF. If larger than this value, it will be cut into smaller pieces before being inputted into DVF.\nvalue is Integer\ndefault is 2099000","category":"page"},{"location":"Running_singleworkflow_params/#DVF_scoreTh","page":"Single workflow parameters","title":"DVF_scoreTh","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"minimum score value calculated by DVF so that contigs are kept by DoViP\t\nvalue is Float (between 0 and 1)\ndefault is 0.7","category":"page"},{"location":"Running_singleworkflow_params/#DVF_pThreshold","page":"Single workflow parameters","title":"DVF_pThreshold","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum p-value calculated by DVF so that contigs are kept by DoViP\t\nvalue is Float (between 0 and 1)\ndefault is 0.01","category":"page"},{"location":"Running_singleworkflow_params/#DVF_p","page":"Single workflow parameters","title":"DVF_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"path/to/dvf.py\t\nmandatory parameter if DVF should be run, no default is given\t","category":"page"},{"location":"Running_singleworkflow_params/#DVF_sbatch_time","page":"Single workflow parameters","title":"DVF_sbatch_time","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"maximum time that a sbatch job running geNomad (in slurm) is allowed to run\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#DVF_cpus_per_task","page":"Single workflow parameters","title":"DVF_cpus_per_task","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum number of CPUs in which DVF should be run (either on a local server or on HPC)\nwhen running on an HPC, this parameter will be further transmited to sbatch\nvalue is Integer\ndefault is 15 ","category":"page"},{"location":"Running_singleworkflow_params/#DVF_sbatch_mem","page":"Single workflow parameters","title":"DVF_sbatch_mem","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum memmory requirements if DVF is run as an sbatch job\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given \t\t","category":"page"},{"location":"Running_singleworkflow_params/#VirSorter2-options","page":"Single workflow parameters","title":"VirSorter2 options","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"If this this step is selected for running (virSorter2_signal=do), the following will happen:  ","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"DoViP will start a VirSorter2 run in the conda environment given by the virSorter2_env parameter, using the following command:\nvirsorter run -i '/path/to/input.fasta' -w 'path/to/output/folder' -j `virSorter2_cpus_per_task` -d `virSorter2DB_p` --min-length `min_contig_length` --min-score `virSorter2_min_score` --tmpdir 'path/to/output/folder/tmp' --viral-gene-required --exclude-lt2gene --hallmark-required-on-short$(hc) --include-groups dsDNAphage,NCLDV,RNA,ssDNA,lavidaviridae --keep-original-seq --rm-tmpdir\n\n# the value of $(hc) in the code above is \" --high-confidence-only\" if the DoViP parameter `virSorter2_high_confidence_only=true` or \"\" if `virSorter2_high_confidence_only=false`\nDoViP will postprocess the VirSorter2 results (final-viral-score.tsv and final-viral-boundary.tsv files) to separate NON-INTEGRATED from INTEGRATED viruses.","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2_signal","page":"Single workflow parameters","title":"virSorter2_signal","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"if VirSorter2 predictions should be considered\npossible values:\n'do' - run this predictor step. Can be always used, regardless of the signals in the previous steps\n'dont' - don't run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a 'dont' or 'remove' signal in the previous step.\n'use'- re-use predictor results calculated in previous runs of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished.\n'use_external' - use results calculated previously by the predictor independently from a DoViP project.\n'ignore' - doesn't take into consideration existing results calculated in a previous run of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished. \n'remove' - removes results calculated in a previous step. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or ignore in the previous step, regardless of their \"progress\". \nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2_res","page":"Single workflow parameters","title":"virSorter2_res","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with VirSorter2 results that were calculated outside DoViP\nneeded only if virSorter2_signal=use_external","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2_env","page":"Single workflow parameters","title":"virSorter2_env","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the conda environment where VirSorter2 is installed\nmandatory parameter if virSorter2 signal is do, no default is given, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2DB_p","page":"Single workflow parameters","title":"virSorter2DB_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with the virSorter2 database\nmandatory parameter if virSorter2 signal is do, no default is given, no default is given\t","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2_high_confidence_only","page":"Single workflow parameters","title":"virSorter2_high_confidence_only","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"will tell VirSorter2 if it should keep only the high confidence contigs\npossible values: 'true' or 'false'\ndefault is 'false'","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2_min_score","page":"Single workflow parameters","title":"virSorter2_min_score","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"mininum score to be used by VirSorter2 to filter viral contigs\nvalue is Float (between 0.0 and 1.0)\ndefault it 0.5","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2_sbatch_time","page":"Single workflow parameters","title":"virSorter2_sbatch_time","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"maximum time that a sbatch job running VirSorter2 (in slurm) is allowed to run\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2_cpus_per_task","page":"Single workflow parameters","title":"virSorter2_cpus_per_task","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum number of CPUs in which VirSorter2 should be run (either on a local server or on HPC)\nwhen running on an HPC, this parameter will be further transmited to sbatch\nvalue is Integer\ndefault is 20 ","category":"page"},{"location":"Running_singleworkflow_params/#virSorter2_sbatch_mem","page":"Single workflow parameters","title":"virSorter2_sbatch_mem","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum memmory requirements if VirSorter2 is run as an sbatch job\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given ","category":"page"},{"location":"Running_singleworkflow_params/#VIBRANT-options","page":"Single workflow parameters","title":"VIBRANT options","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"If this this step is selected for running (vibrant_signal=do), the following will happen:  ","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"DoViP will start a VIBRANT run in the conda environment given by the vibrant_env parameter, using the following command:\npython3 `vibrant_p` -f nucl -i '/path/to/input.fasta' -folder 'path/to/output/folder' -t `vibrant_cpus_per_task` -d `vibrantDB_p` -l `min_contig_length`\nDoViP will postprocess the VIBRANT outputs (VIBRANT_integrated_prophage_coordinates_$(sampleName).tsv, $(sampleName).phages_lytic.fna) to separate NON-INTEGRATED from INTEGRATED viruses.","category":"page"},{"location":"Running_singleworkflow_params/#vibrant_signal","page":"Single workflow parameters","title":"vibrant_signal","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"if VIBRANT predictions should be considered\npossible values:\n'do' - run this predictor step. Can be always used, regardless of the signals in the previous steps\n'dont' - don't run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a 'dont' or 'remove' signal in the previous step.\n'use'- re-use predictor results calculated in previous runs of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished.\n'use_external' - use results calculated previously by the predictor independently from a DoViP project.\n'ignore' - doesn't take into consideration existing results calculated in a previous run of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished. \n'remove' - removes results calculated in a previous step. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or ignore in the previous step, regardless of their \"progress\". \nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#vibrant_res","page":"Single workflow parameters","title":"vibrant_res","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with VIBRANT results that were calculated outside DoViP\nneeded only if vibrant_signal=use_external","category":"page"},{"location":"Running_singleworkflow_params/#vibrant_env","page":"Single workflow parameters","title":"vibrant_env","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the conda environment where VIBRANT is installed\nmandatory parameter if VIBRANT signal is do, no default is given, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#vibrant_p","page":"Single workflow parameters","title":"vibrant_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"path/to/VIBRANT_run.py\t\nmandatory parameter if VIBRANT signal is do, no default is given, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#vibrantDB_p","page":"Single workflow parameters","title":"vibrantDB_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with the VIBRANT database\nmandatory parameter if VIBRANT signal is do, no default is given, no default is given\t","category":"page"},{"location":"Running_singleworkflow_params/#vibrant_sbatch_time","page":"Single workflow parameters","title":"vibrant_sbatch_time","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"maximum time that a sbatch job running VIBRANT (in slurm) is allowed to run\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC and VIBRANT signal is do, no default is given ","category":"page"},{"location":"Running_singleworkflow_params/#vibrant_cpus_per_task","page":"Single workflow parameters","title":"vibrant_cpus_per_task","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum number of CPUs in which VIBRANT should be run (either on a local server or on HPC)\nwhen running on an HPC, this parameter will be further transmited to sbatch\nvalue is Integer\ndefault is 20 ","category":"page"},{"location":"Running_singleworkflow_params/#vibrant_sbatch_mem","page":"Single workflow parameters","title":"vibrant_sbatch_mem","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum memmory requirements if VIBRANT is run as an sbatch job\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC and VIBRANT signal is do, no default is given ","category":"page"},{"location":"Running_singleworkflow_params/#ViralVerify-options","page":"Single workflow parameters","title":"ViralVerify options","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"If this this step is selected for running (viralVerify_signal=do), the following will happen:  ","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"DoViP will start a ViralVerify run in the conda environment given by the viralVerify_env parameter, using the following command:\nviralVerify_p` -f '/path/to/input.fasta' -o 'path/to/output/folder' --hmm `viralVerifyDB_p` --thr `viralVerfify_threshold` -t `viralVerify_cpus_per_task` -p\nDoViP will postprocess ViralVerify outputs (_result_table.csv) to select only the viral contigs with a score bigger than viralVerfify_threshold.  Only NON-INTEGRATED viruses are outputed by ViralVerify.","category":"page"},{"location":"Running_singleworkflow_params/#viralVerify_signal","page":"Single workflow parameters","title":"viralVerify_signal","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"if ViralVerify predictions should be considered\npossible values:\n'do' - run this predictor step. Can be always used, regardless of the signals in the previous steps\n'dont' - don't run this predictor step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a 'dont' or 'remove' signal in the previous step.\n'use'- re-use predictor results calculated in previous runs of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished.\n'use_external' - use results calculated previously by the predictor independently from a DoViP project.\n'ignore' - doesn't take into consideration existing results calculated in a previous run of this project. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished. \n'remove' - removes results calculated in a previous step. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or ignore in the previous step, regardless of their \"progress\". \nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#viralVerify_res","page":"Single workflow parameters","title":"viralVerify_res","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with ViralVerify results that were calculated outside DoViP\nneeded only if viralVerify_signal=use_external","category":"page"},{"location":"Running_singleworkflow_params/#viralVerify_env","page":"Single workflow parameters","title":"viralVerify_env","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the conda environment where ViralVerify is installed\nmandatory parameter if ViralVerify signal is do, no default is given, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#viralVerifyDB_p","page":"Single workflow parameters","title":"viralVerifyDB_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the HMM profile file of the ViralVerify database\nmandatory parameter if ViralVerify signal is do, no default is given, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#viralVerify_p","page":"Single workflow parameters","title":"viralVerify_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"path/to/bin/viralverify\t\nmandatory parameter if ViralVerify signal is do, no default is given, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#viralVerfify_threshold","page":"Single workflow parameters","title":"viralVerfify_threshold","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"score used to select viral contigs outputted by ViralVerify. \nvalue is Integer (between 1 and 1000)\ndefault is 7","category":"page"},{"location":"Running_singleworkflow_params/#viralVerify_sbatch_time","page":"Single workflow parameters","title":"viralVerify_sbatch_time","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"maximum time that a sbatch job running ViralVeriffy (in slurm) is allowed to run\nmandatory parameter when running on HPC and ViralVerify signal is do, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#viralVerify_cpus_per_task","page":"Single workflow parameters","title":"viralVerify_cpus_per_task","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum number of CPUs in which ViralVerify should be run (either on a local server or on HPC)\nvalue is Integer\ndefault is 20 ","category":"page"},{"location":"Running_singleworkflow_params/#viralVerify_sbatch_mem","page":"Single workflow parameters","title":"viralVerify_sbatch_mem","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum memmory requirements if ViralVerify is run as an sbatch job\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC and ViralVerify signal is do, no default is given ","category":"page"},{"location":"Running_singleworkflow_params/#PARAMETERS-FOR-CONSENSUS-PREDICTION-STEPS","page":"Single workflow parameters","title":"PARAMETERS FOR CONSENSUS PREDICTION STEPS","text":"","category":"section"},{"location":"Running_singleworkflow_params/#CheckV-options","page":"Single workflow parameters","title":"CheckV options","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"CheckV is used both on the NON-INTEGRATED and INTEGRATED viruses branch to detect viral contigs with host contamination (if they are on the NON-INTEGRATED branch they will be moved to the INTEGRATED branch) and remove the host regions.   In all cases, CheckV is run with the following command:  ","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"checkv end_to_end 'path/to/input.fasta' 'path/to/output_folder' -t `checkv_cpus_per_task` -d `checkvDB_p` --remove_tmp","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"The CheckV statistics will be incorporated in the final virus contig tables.","category":"page"},{"location":"Running_singleworkflow_params/#checkv_env","page":"Single workflow parameters","title":"checkv_env","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the conda environment where CheckV is installed\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#checkvDB_p","page":"Single workflow parameters","title":"checkvDB_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with the CheckV database\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#checkv_sbatch_time","page":"Single workflow parameters","title":"checkv_sbatch_time","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"maximum time that a sbatch job running CheckV (in slurm) is allowed to run\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#checkv_cpus_per_task","page":"Single workflow parameters","title":"checkv_cpus_per_task","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum number of CPUs in which CheckV should be run (either on a local server or on HPC)\nwhen running on an HPC, this parameter will be further transmited to sbatch\nvalue is Integer\ndefault is 2 ","category":"page"},{"location":"Running_singleworkflow_params/#checkv_sbatch_mem","page":"Single workflow parameters","title":"checkv_sbatch_mem","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum memmory requirements if checkV is run as an sbatch job\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given ","category":"page"},{"location":"Running_singleworkflow_params/#PhaTYP-options","page":"Single workflow parameters","title":"PhaTYP options","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"PhaTYP is used on the NON-INTEGRATED viruses branch to predict the potential lifestyle of the respective viruses.   PhaTYP is run with the following command:","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"python `phaTYP_p/PhaTYP_single.py` --contigs 'path/to/input.fasta' --threads `phaTYP_cpus_per_task` --len `min_contig_length` --rootpth 'path/to/temp_output_folder' --out 'path/to/output_folder' --dbdir `phaTYPdb_p` --parampth `phaTYPparam_p`","category":"page"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"The PhaTYP predictions will be incorporated in the final virus contig tables.","category":"page"},{"location":"Running_singleworkflow_params/#phaTYP_env","page":"Single workflow parameters","title":"phaTYP_env","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the conda environment where PhaTYP is installed\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#phaTYP_p","page":"Single workflow parameters","title":"phaTYP_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with PhaTYP\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#phaTYPdb_p","page":"Single workflow parameters","title":"phaTYPdb_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with the PhaTYP database\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#phaTYPparam_p","page":"Single workflow parameters","title":"phaTYPparam_p","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the path toward the folder with the PhaTYP parameters (see PhaTYP manual)\nmandatory parameter, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#phaTYP_sbatch_time","page":"Single workflow parameters","title":"phaTYP_sbatch_time","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"maximum time that a sbatch job running PhaTYP (in slurm) is allowed to run\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given","category":"page"},{"location":"Running_singleworkflow_params/#phaTYP_cpus_per_task","page":"Single workflow parameters","title":"phaTYP_cpus_per_task","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum number of CPUs in which PhaTYP should be run (either on a local server or on HPC)\nwhen running on an HPC, this parameter will be further transmited to sbatch\nvalue is Integer\ndefault is 2 ","category":"page"},{"location":"Running_singleworkflow_params/#phaTYP_sbatch_mem","page":"Single workflow parameters","title":"phaTYP_sbatch_mem","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"the maximum memmory requirements if PhaTYP is run as an sbatch job\nthis parameter will be further transmited to sbatch\nmandatory parameter when running on HPC, no default is given ","category":"page"},{"location":"Running_singleworkflow_params/#NON-INTEGRATED-viruses-branch:-options-to-select-final-viral-contigs","page":"Single workflow parameters","title":"NON-INTEGRATED viruses branch: options to select final viral contigs","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"At the end of the workflow DoViP keeps only the NON-INTEGRATED viral contigs that fullfill certain criteria. The selection depends on the method used by CheckV to estimate the viral contig completeness, on the value of the viral contig completeness, and on the number of initial predictors that outputed the respective viral contig. ","category":"page"},{"location":"Running_singleworkflow_params/#If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:","page":"Single workflow parameters","title":"If CheckV could not estimate the completeness of the viral contigs:","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"it means that CheckV could not find similarities in its database\nthan only the number of viral predictors is taken into consideration. \nI recommend using a higher number of initial predictors, to increase the confidence that the respective contig is actually viral  ","category":"page"},{"location":"Running_singleworkflow_params/#NONInt_th_num_predictors_CheckV_NA","page":"Single workflow parameters","title":"NONInt_th_num_predictors_CheckV_NA","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameters gives the minimum number of initial predictors necessary to keep a viral contig if completeness is NA\nvalue is Integer\ndefault is 3","category":"page"},{"location":"Running_singleworkflow_params/#If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:","page":"Single workflow parameters","title":"If CheckV used AAIHighConf method to estimate the completeness of the viral contigs:","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"it means that it found related viruses in its database and it can estimate the completeness with high precission\nall viral contigs with a completeness of at least 90% are kept, regardless of the number of initial predictors ","category":"page"},{"location":"Running_singleworkflow_params/#NONInt_th_num_predictors_CheckV_AAIHighConf","page":"Single workflow parameters","title":"NONInt_th_num_predictors_CheckV_AAIHighConf","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is AAIHighConf\nvalue is Integer\ndefault is 1","category":"page"},{"location":"Running_singleworkflow_params/#NONInt_th_completeness_CheckV_AAIHighConf","page":"Single workflow parameters","title":"NONInt_th_completeness_CheckV_AAIHighConf","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is AAIHighConf\nvalue is Float\ndefault is 30.0","category":"page"},{"location":"Running_singleworkflow_params/#If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:","page":"Single workflow parameters","title":"If CheckV used AAIMediumConf method to estimate the completeness of the viral contigs:","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"it means that it found related viruses in its database and it can estimate the completeness with moderate precission","category":"page"},{"location":"Running_singleworkflow_params/#NONInt_th_num_predictors_CheckV_AAIMediumConf","page":"Single workflow parameters","title":"NONInt_th_num_predictors_CheckV_AAIMediumConf","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is AAIMediumConf\nvalue is Integer\ndefault is 2\t","category":"page"},{"location":"Running_singleworkflow_params/#NONInt_th_completeness_CheckV_AAIMediumConf","page":"Single workflow parameters","title":"NONInt_th_completeness_CheckV_AAIMediumConf","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is AAIMediumConf\nvalue is Float\ndefault is 10.0","category":"page"},{"location":"Running_singleworkflow_params/#If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:","page":"Single workflow parameters","title":"If CheckV used HMM method to estimate the completeness of the viral contigs:","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"it means that it found distantly related viruses in its database and it can estimate the completeness with low precission","category":"page"},{"location":"Running_singleworkflow_params/#NONInt_th_num_predictors_CheckV_HMM","page":"Single workflow parameters","title":"NONInt_th_num_predictors_CheckV_HMM","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is HMM\nvalue is Integer\ndefault is 2\t","category":"page"},{"location":"Running_singleworkflow_params/#NONInt_th_completeness_CheckV_HMM","page":"Single workflow parameters","title":"NONInt_th_completeness_CheckV_HMM","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is HMM\nvalue is Float\ndefault is 10.0","category":"page"},{"location":"Running_singleworkflow_params/#INTEGRATED-viruses-branch:-options-to-select-final-viral-contigs","page":"Single workflow parameters","title":"INTEGRATED viruses branch: options to select final viral contigs","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"At the end of the workflow DoViP keeps only the INTEGRATED viruses that fullfill certain criteria. The selection depends on the method used by CheckV to estimate the viral contig completeness, on the value of the viral contig completeness, and on the number of initial predictors that outputed the respective viral contig.   ","category":"page"},{"location":"Running_singleworkflow_params/#If-CheckV-could-not-estimate-the-completeness-of-the-viral-contigs:-2","page":"Single workflow parameters","title":"If CheckV could not estimate the completeness of the viral contigs:","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"it means that CheckV could not find similarities in its database\nthan only the number of viral predictors is taken into consideration. \nI recommend using a higher number of initial predictors, to increase the confidence that the respective contig is actually viral  ","category":"page"},{"location":"Running_singleworkflow_params/#Int_th_num_predictors_CheckV_NA","page":"Single workflow parameters","title":"Int_th_num_predictors_CheckV_NA","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum number of initial predictors necessary to keep a viral contig if completeness is NA\nvalue is Integer\ndefault is 3","category":"page"},{"location":"Running_singleworkflow_params/#If-CheckV-used-AAIHighConf-method-to-estimate-the-completeness-of-the-viral-contigs:-2","page":"Single workflow parameters","title":"If CheckV used AAIHighConf method to estimate the completeness of the viral contigs:","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"it means that it found related viruses in its database and it can estimate the completeness with high precission\nall viral contigs with a completeness of at least 90% are kept, regardless of the number of initial predictors ","category":"page"},{"location":"Running_singleworkflow_params/#Int_th_num_predictors_CheckV_AAIHighConf","page":"Single workflow parameters","title":"Int_th_num_predictors_CheckV_AAIHighConf","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is AAIHighConf\nvalue is Integer\ndefault is 1","category":"page"},{"location":"Running_singleworkflow_params/#Int_th_completeness_CheckV_AAIHighConf","page":"Single workflow parameters","title":"Int_th_completeness_CheckV_AAIHighConf","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is AAIHighConf\nvalue is Float\ndefault is 30.0","category":"page"},{"location":"Running_singleworkflow_params/#If-CheckV-used-AAIMediumConf-method-to-estimate-the-completeness-of-the-viral-contigs:-2","page":"Single workflow parameters","title":"If CheckV used AAIMediumConf method to estimate the completeness of the viral contigs:","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"it means that it found related viruses in its database and it can estimate the completeness with moderate precission","category":"page"},{"location":"Running_singleworkflow_params/#Int_th_num_predictors_CheckV_AAIMediumConf","page":"Single workflow parameters","title":"Int_th_num_predictors_CheckV_AAIMediumConf","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is AAIMediumConf\nvalue is Integer\ndefault is 2\t","category":"page"},{"location":"Running_singleworkflow_params/#Int_th_completeness_CheckV_AAIMediumConf","page":"Single workflow parameters","title":"Int_th_completeness_CheckV_AAIMediumConf","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is AAIMediumConf\nvalue is Float\ndefault is 10.0","category":"page"},{"location":"Running_singleworkflow_params/#If-CheckV-used-HMM-method-to-estimate-the-completeness-of-the-viral-contigs:-2","page":"Single workflow parameters","title":"If CheckV used HMM method to estimate the completeness of the viral contigs:","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"it means that it found distantly related viruses in its database and it can estimate the completeness with low precission","category":"page"},{"location":"Running_singleworkflow_params/#Int_th_num_predictors_CheckV_HMM","page":"Single workflow parameters","title":"Int_th_num_predictors_CheckV_HMM","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum number of initial predictors necessary to keep a viral contig if the method for determining the completeness is HMM\nvalue is Integer\ndefault is 2\t\t","category":"page"},{"location":"Running_singleworkflow_params/#Int_th_completeness_CheckV_HMM","page":"Single workflow parameters","title":"Int_th_completeness_CheckV_HMM","text":"","category":"section"},{"location":"Running_singleworkflow_params/","page":"Single workflow parameters","title":"Single workflow parameters","text":"this parameter gives the minimum completeness to keep a viral contig when the method for determining the completeness is HMM\nvalue is Float\ndefault is 10.0","category":"page"},{"location":"#DoViP-Do-Virus-Prediction","page":"Home","title":"DoViP - Do Virus Prediction","text":"","category":"section"},{"location":"Installation/","page":"Overview","title":"Overview","text":"DoViP runs and then integrates the results from several virus analysis tools. These tools come with their own conda environments and databases. It is quite likely that the users have already installed these tools on their servers. As a result, I have decided not to include everything in a big conda environment, but rather to keep each tool in its own conda environment, separated from the conda environment of DoViP.","category":"page"}]
}
