var documenterSearchIndex = {"docs":
[{"location":"installation_conda_dependencies/#DoViP-dependencies-and-their-installation","page":"Dependencies","title":"DoViP dependencies and their installation","text":"","category":"section"},{"location":"installation_conda_dependencies/#geNomad","page":"Dependencies","title":"geNomad","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"geNomad use and installation is described in detail here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"mamba create -p path/to/conda_genomad -c conda-forge -c bioconda genomad\n\n# database setup\nconda activate conda_genomad\ncd /path/for/databases\ngenomad download-database .","category":"page"},{"location":"installation_conda_dependencies/#VIBRANT","page":"Dependencies","title":"VIBRANT","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"VIBRANT use and installation is described in detail here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# env\nconda create -n conda_VIBRANT\nconda activate conda_VIBRANT\n\n# dependencies\nconda install -c bioconda hmmer\nconda install -c bioconda prodigal\nconda install -c bioconda biopython\nconda install -c anaconda pandas\nconda install matplotlib\nconda install seaborn\nconda install -c anaconda numpy\nconda install scikit-learn\n\n# install VIBRANT itself\ncd path/for/vibrant/repo\ngit clone https://github.com/AnantharamanLab/VIBRANT\nchmod -R 777 VIBRANT/\n\n# database\ncd databases\npip install --upgrade scikit-learn==0.21.3\npip install --upgrade numpy==1.17.0\npython3 VIBRANT_setup.py ","category":"page"},{"location":"installation_conda_dependencies/#VirSorter2","page":"Dependencies","title":"VirSorter2","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"VirSorter 2 use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# create env and instal, virsorter2\nconda create -p path/to/conda_virsorter2\nmamba activate path/to/conda_virsorter2\nmamba install -c conda-forge -c bioconda virsorter=2\n\n# databases\ncd conda_virsorter2/\nvirsorter setup -d db -j 4","category":"page"},{"location":"installation_conda_dependencies/#DeepVir-Finder","page":"Dependencies","title":"DeepVir Finder","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"DeepVir Finder use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# create env\nconda create -p /path/to/conda_DVF python=3.6 numpy theano=1.0.3 keras=2.2.4 scikit-learn Biopython h5py\nconda activate conda_DVF\n\n# clone the git repo\ncd path/for/repo\ngit clone https://github.com/jessieren/DeepVirFinder","category":"page"},{"location":"installation_conda_dependencies/#CheckV","page":"Dependencies","title":"CheckV","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"CheckV use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# env\nconda create -p path/to/conda_checkV\nconda activate conda_checkV\n\n#\nconda install -c conda-forge -c bioconda checkV\n\n\n# database\ncd path/for/databases/\ncheckv download_database ./","category":"page"},{"location":"installation_conda_dependencies/#PhaBox","page":"Dependencies","title":"PhaBox","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"PhaBox use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"cd path/for/repo\ngit clone https://github.com/KennthShang/PhaBOX\ncd PhaBOX/\nconda env create -f webserver.yml -p path/for/conda_PhaBOX\nconda activate conda_PhaBOX\n\n# databases\npip install gdown\ngdown  --id 1hjACPsIOqqcS5emGaduYvYrCzrIpt2_9\ngdown  --id 1E94ii3Q0O8ZBm7UsyDT_n06YekNtfV20\n\npath/for/repo/unzip phagesuite_database.zip  > /dev/null\npath/for/repo/unzip phagesuite_parameters.zip  > /dev/null\n\n\ncp blastxml_to_tabular.pypath/for/conda_PhaBOX/bin/blastxml_to_tabular.py\nchmod 777 path/for/conda_PhaBOX/bin/blastxml_to_tabular.py","category":"page"},{"location":"installation_DoViP_env/#Variant-1:-Step-by-step","page":"DoViP environment","title":"Variant 1: Step by step","text":"","category":"section"},{"location":"installation_DoViP_env/#1.-Create-the-conda-environment-for-DoViP-and-install-Julia-programming-language","page":"DoViP environment","title":"1. Create the conda environment for DoViP and install Julia programming language","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"# create conda env\nconda create -p /path/toward/conda_DoViP\nconda activate /path/toward/conda_DoViP\n\n# download and install julia version 1.10.0\ncd /path/toward/conda_DoViP\nwget https://julialang-s3.julialang.org/bin/linux/x64/1.10/julia-1.10.0-linux-x86_64.tar.gz\ntar -xzvf julia-1.10.0-linux-x86_64.tar.gz\nrm julia-1.10.0-linux-x86_64.tar.gz\nmkdir bin\n\n# make a symbolic link of the julia executable in the /path/toward/conda_DoViP/bin/ folder\nln -s /path/toward/conda_DoViP/julia-1.10.0/bin/julia /path/toward/conda_DoViP/bin/julia\n","category":"page"},{"location":"installation_DoViP_env/#2.-Bring-the-DoViP-repo","page":"DoViP environment","title":"2. Bring the DoViP repo","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"cd /path/toward/conda_DoViP\ngit clone https://github.com/CristinaMoraru/DoViP.git\n\nmv DoViP/* .\nrm DoViP\n\n# give permissions to the folder where julia saved its libraries\ncd share\nchmod 777 -R julia","category":"page"},{"location":"installation_DoViP_env/#3.-Refresh-the-DoViP-conda-environment","page":"DoViP environment","title":"3. Refresh the DoViP conda environment","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"conda deactivate conda_DoViP\nconda activate /path/toward/conda_DoViP\n","category":"page"},{"location":"installation_DoViP_env/#4.-Instantiate-DoViP-libraries","page":"DoViP environment","title":"4. Instantiate DoViP libraries","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"\ncd /path/toward/conda_DoViP/\n\njulia -e 'using Pkg; Pkg.activate(\"DoViP_App.jl\"); Pkg.instantiate()' \n","category":"page"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"Instantiation is necessary to bring on your machine the packages on which DoViP depends. After instantiation, it will attempt to start DoViP already, and therefore you will see an error message:  \"\". Ignore it.","category":"page"},{"location":"installation_DoViP_env/#5.Test-installation","page":"DoViP environment","title":"5.Test installation","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"DoViP --help","category":"page"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"You should get the following message: \"DoViP - a workflow for virus prediction in metagenomes.\"","category":"page"},{"location":"installation_DoViP_env/#Variant-2:-From-a-conda-package","page":"DoViP environment","title":"Variant 2: From a conda package","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"to be prepared","category":"page"},{"location":"Running/#DoViP-parameters","page":"Running","title":"DoViP parameters","text":"","category":"section"},{"location":"Running/#Arguments-to-be-given-to-DoViP-through-the-.tsv-file","page":"Running","title":"Arguments to be given to DoViP through the .tsv file","text":"","category":"section"},{"location":"Running/#General-project-options","page":"Running","title":"General project options","text":"","category":"section"},{"location":"Running/#inref","page":"Running","title":"inref","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the path toward the fasta file containing the input contigs. \n* Accepted file extensions are: \".fasta\", \".fna\", \".fst\", \".fas\", \".fa\", \".mfa\"\n* mandatory parameter, no default is given","category":"page"},{"location":"Running/#projtype","page":"Running","title":"projtype","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the only acceptable value here is 'singleworkflow', and it tells DoViP that only one metagenome is analysed for this row.\n* mandatory parameter, no default is given","category":"page"},{"location":"Running/#sample_set","page":"Running","title":"sample_set","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the name of the sample/metagenome collection\n* mandatory parameter, no default is given","category":"page"},{"location":"Running/#use_slurm","page":"Running","title":"use_slurm","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* if calculations should be run on an High Performance Computing Cluster, using slurm \n* possible values: 'true' or 'false'\n* default is 'false'","category":"page"},{"location":"Running/#continue","page":"Running","title":"continue","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* if the calculations for this sample should be continued from a previous run of the project\n* this is useful for example when changing the desired initial predictors (see details below) or the thresholds for the final selection of the viral contigs\n* possible values: 'true' or 'false'\n* default is 'false'\n    Note: if continue is set to 'false' and the project folder aldready exists, a message will appear asking to confirm or deny the deletion of the existing folder.","category":"page"},{"location":"Running/#stop_after_initial_predictors","page":"Running","title":"stop_after_initial_predictors","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* if the calculations should stop after the initial predictions have finished (useful for example if the project should be continued later)\n* possible values: 'true' or 'false'\n* default is 'false'","category":"page"},{"location":"Running/#min_contig_length","page":"Running","title":"min_contig_length","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the minimum length of the contigs from the input file (inref) which should be kept\n* only contigs equal or larger than this value will be sent to the intial predictors\n* value is Integer\n* default value is 1000","category":"page"},{"location":"Running/#INITIAL-PREDICTIONS","page":"Running","title":"INITIAL PREDICTIONS","text":"","category":"section"},{"location":"Running/#geNomad-options","page":"Running","title":"geNomad options","text":"","category":"section"},{"location":"Running/#genomad_signal","page":"Running","title":"genomad_signal","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* if genomad predictions should be considered\n* possible values:\n    * 'do' - run the step. Can be always used, regardless of the signals in the previous steps\n    * 'dont' - don't run the step. It can be used only if the project is new (continue = false), or, for continued projects, if there was a 'dont' or 'remove' signal in the previous step.\n    * 'use'  - doesn't take into cosideration existing results. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished.\n    * 'use_external' - use results calculate previously independent from a DoViP project.\n    * 'ignore' - doesn't take into cosideration existing results. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or 'ignore' in the previous step, and, if their respective \"progress\" is finished. \n    * 'remove' - removes results calculated in a previous step. It can't be used if the project is new (continue = false). For continued projects, it can be used only after 'do', 'use' or ignore in the previous step, regardless of their \"progress\". \n    * mandatory parameter, no default is given","category":"page"},{"location":"Running/#genomad_res","page":"Running","title":"genomad_res","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the path toward the folder with geNomad results that were calculated outside DoViP\n* it will only have effect ig 'genomad_signal=use_external'","category":"page"},{"location":"Running/#genomad_env","page":"Running","title":"genomad_env","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the path toward the conda environment where geNomad was installed\n* mandatory parameter, no default is given","category":"page"},{"location":"Running/#genomadDB_p","page":"Running","title":"genomadDB_p","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the path toward the folder with the geNomad database\n* mandatory parameter, no default is given","category":"page"},{"location":"Running/#genomad_min_score","page":"Running","title":"genomad_min_score","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* mininum score to be used by geNomad\n* value is Float\n* default it 0.7","category":"page"},{"location":"Running/#genomad_sbatch_time","page":"Running","title":"genomad_sbatch_time","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* maximum time that a sbatch job running geNomad (in slurm) should have\n* mandatory parameter when running on HPC, no default is given","category":"page"},{"location":"Running/#genomad_cpus_per_task","page":"Running","title":"genomad_cpus_per_task","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the maximum number of CPUs in which geNomad should be run (either on a local server or on HPC)\n* value is Integer\n* default is 20","category":"page"},{"location":"Running/#genomad_sbatch_mem","page":"Running","title":"genomad_sbatch_mem","text":"","category":"section"},{"location":"Running/","page":"Running","title":"Running","text":"* the maximum memmory requirements if geNomad is run as an sbatch job\n* mandatory parameter when running on HPC, no default is given","category":"page"},{"location":"Running/#DeepVir-Finder-options","page":"Running","title":"DeepVir Finder options","text":"","category":"section"},{"location":"Running/#DVF_signal","page":"Running","title":"DVF_signal","text":"","category":"section"},{"location":"Running/#DVF_env","page":"Running","title":"DVF_env","text":"","category":"section"},{"location":"Running/#DVF_maxContigLen","page":"Running","title":"DVF_maxContigLen","text":"","category":"section"},{"location":"Running/#DVF_scoreTh","page":"Running","title":"DVF_scoreTh","text":"","category":"section"},{"location":"Running/#DVF_pThreshold","page":"Running","title":"DVF_pThreshold","text":"","category":"section"},{"location":"Running/#DVF_p","page":"Running","title":"DVF_p","text":"","category":"section"},{"location":"Running/#DVF_sbatch_time","page":"Running","title":"DVF_sbatch_time","text":"","category":"section"},{"location":"Running/#DVF_cpus_per_task","page":"Running","title":"DVF_cpus_per_task","text":"","category":"section"},{"location":"Running/#DVF_sbatch_mem","page":"Running","title":"DVF_sbatch_mem","text":"","category":"section"},{"location":"Running/#VirSorter2-options","page":"Running","title":"VirSorter2 options","text":"","category":"section"},{"location":"Running/#virSorter2_signal","page":"Running","title":"virSorter2_signal","text":"","category":"section"},{"location":"Running/#virSorter2_res","page":"Running","title":"virSorter2_res","text":"","category":"section"},{"location":"Running/#virSorter2_env","page":"Running","title":"virSorter2_env","text":"","category":"section"},{"location":"Running/#virSorter2DB_p","page":"Running","title":"virSorter2DB_p","text":"","category":"section"},{"location":"Running/#virSorter2_high_confidence_only","page":"Running","title":"virSorter2_high_confidence_only","text":"","category":"section"},{"location":"Running/#virSorter2_min_score","page":"Running","title":"virSorter2_min_score","text":"","category":"section"},{"location":"Running/#virSorter2_sbatch_time","page":"Running","title":"virSorter2_sbatch_time","text":"","category":"section"},{"location":"Running/#virSorter2_cpus_per_task","page":"Running","title":"virSorter2_cpus_per_task","text":"","category":"section"},{"location":"Running/#virSorter2_sbatch_mem","page":"Running","title":"virSorter2_sbatch_mem","text":"","category":"section"},{"location":"Running/#VIBRANT-options","page":"Running","title":"VIBRANT options","text":"","category":"section"},{"location":"Running/#vibrant_signal","page":"Running","title":"vibrant_signal","text":"","category":"section"},{"location":"Running/#vibrant_res","page":"Running","title":"vibrant_res","text":"","category":"section"},{"location":"Running/#vibrant_env","page":"Running","title":"vibrant_env","text":"","category":"section"},{"location":"Running/#vibrant_p","page":"Running","title":"vibrant_p","text":"","category":"section"},{"location":"Running/#vibrantDB_p","page":"Running","title":"vibrantDB_p","text":"","category":"section"},{"location":"Running/#vibrant_sbatch_time","page":"Running","title":"vibrant_sbatch_time","text":"","category":"section"},{"location":"Running/#vibrant_cpus_per_task","page":"Running","title":"vibrant_cpus_per_task","text":"","category":"section"},{"location":"Running/#vibrant_sbatch_mem","page":"Running","title":"vibrant_sbatch_mem","text":"","category":"section"},{"location":"Running/#ViralVerify-options","page":"Running","title":"ViralVerify options","text":"","category":"section"},{"location":"Running/#viralVerify_signal","page":"Running","title":"viralVerify_signal","text":"","category":"section"},{"location":"Running/#viralVerify_res","page":"Running","title":"viralVerify_res","text":"","category":"section"},{"location":"Running/#viralVerify_env","page":"Running","title":"viralVerify_env","text":"","category":"section"},{"location":"Running/#viralVerifyDB_p","page":"Running","title":"viralVerifyDB_p","text":"","category":"section"},{"location":"Running/#viralVerify_p","page":"Running","title":"viralVerify_p","text":"","category":"section"},{"location":"Running/#viralVerfify_threshold","page":"Running","title":"viralVerfify_threshold","text":"","category":"section"},{"location":"Running/#viralVerify_sbatch_time","page":"Running","title":"viralVerify_sbatch_time","text":"","category":"section"},{"location":"Running/#viralVerify_cpus_per_task","page":"Running","title":"viralVerify_cpus_per_task","text":"","category":"section"},{"location":"Running/#viralVerify_sbatch_mem","page":"Running","title":"viralVerify_sbatch_mem","text":"","category":"section"},{"location":"Running/#CheckV-options","page":"Running","title":"CheckV options","text":"","category":"section"},{"location":"Running/#checkv_env","page":"Running","title":"checkv_env","text":"","category":"section"},{"location":"Running/#checkvDB_p","page":"Running","title":"checkvDB_p","text":"","category":"section"},{"location":"Running/#checkv_sbatch_time","page":"Running","title":"checkv_sbatch_time","text":"","category":"section"},{"location":"Running/#checkv_cpus_per_task","page":"Running","title":"checkv_cpus_per_task","text":"","category":"section"},{"location":"Running/#checkv_sbatch_mem","page":"Running","title":"checkv_sbatch_mem","text":"","category":"section"},{"location":"Running/#PhaTYP-options","page":"Running","title":"PhaTYP options","text":"","category":"section"},{"location":"Running/#phaTYP_env","page":"Running","title":"phaTYP_env","text":"","category":"section"},{"location":"Running/#phaTYP_p","page":"Running","title":"phaTYP_p","text":"","category":"section"},{"location":"Running/#phaTYPdb_p","page":"Running","title":"phaTYPdb_p","text":"","category":"section"},{"location":"Running/#phaTYPparam_p","page":"Running","title":"phaTYPparam_p","text":"","category":"section"},{"location":"Running/#phaTYP_sbatch_time","page":"Running","title":"phaTYP_sbatch_time","text":"","category":"section"},{"location":"Running/#phaTYP_cpus_per_task","page":"Running","title":"phaTYP_cpus_per_task","text":"","category":"section"},{"location":"Running/#phaTYP_sbatch_mem","page":"Running","title":"phaTYP_sbatch_mem","text":"","category":"section"},{"location":"Running/#NON-INTEGRATED-viruses-branch:-options-to-select-final-viral-contigs","page":"Running","title":"NON-INTEGRATED viruses branch: options to select final viral contigs","text":"","category":"section"},{"location":"Running/#NONInt_th_num_predictors_CheckV_NA","page":"Running","title":"NONInt_th_num_predictors_CheckV_NA","text":"","category":"section"},{"location":"Running/#NONInt_th_num_predictors_CheckV_AAIHighConf","page":"Running","title":"NONInt_th_num_predictors_CheckV_AAIHighConf","text":"","category":"section"},{"location":"Running/#NONInt_th_completeness_CheckV_AAIHighConf","page":"Running","title":"NONInt_th_completeness_CheckV_AAIHighConf","text":"","category":"section"},{"location":"Running/#NONInt_th_num_predictors_CheckV_AAIMediumConf","page":"Running","title":"NONInt_th_num_predictors_CheckV_AAIMediumConf","text":"","category":"section"},{"location":"Running/#NONInt_th_completeness_CheckV_AAIMediumConf","page":"Running","title":"NONInt_th_completeness_CheckV_AAIMediumConf","text":"","category":"section"},{"location":"Running/#NONInt_th_num_predictors_CheckV_HMM","page":"Running","title":"NONInt_th_num_predictors_CheckV_HMM","text":"","category":"section"},{"location":"Running/#NONInt_th_completeness_CheckV_HMM","page":"Running","title":"NONInt_th_completeness_CheckV_HMM","text":"","category":"section"},{"location":"Running/#INTEGRATED-viruses-branch:-options-to-select-final-viral-contigs","page":"Running","title":"INTEGRATED viruses branch: options to select final viral contigs","text":"","category":"section"},{"location":"Running/#Int_th_num_predictors_CheckV_NA","page":"Running","title":"Int_th_num_predictors_CheckV_NA","text":"","category":"section"},{"location":"Running/#Int_th_num_predictors_CheckV_AAIHighConf","page":"Running","title":"Int_th_num_predictors_CheckV_AAIHighConf","text":"","category":"section"},{"location":"Running/#Int_th_completeness_CheckV_AAIHighConf","page":"Running","title":"Int_th_completeness_CheckV_AAIHighConf","text":"","category":"section"},{"location":"Running/#Int_th_num_predictors_CheckV_AAIMediumConf","page":"Running","title":"Int_th_num_predictors_CheckV_AAIMediumConf","text":"","category":"section"},{"location":"Running/#Int_th_completeness_CheckV_AAIMediumConf","page":"Running","title":"Int_th_completeness_CheckV_AAIMediumConf","text":"","category":"section"},{"location":"Running/#Int_th_num_predictors_CheckV_HMM","page":"Running","title":"Int_th_num_predictors_CheckV_HMM","text":"","category":"section"},{"location":"Running/#Int_th_completeness_CheckV_HMM","page":"Running","title":"Int_th_completeness_CheckV_HMM","text":"","category":"section"},{"location":"#DoViP-Do-Virus-Prediction","page":"Home","title":"DoViP - Do Virus Prediction","text":"","category":"section"},{"location":"Installation/","page":"Overview","title":"Overview","text":"DoViP runs and then integrates the results from several virus analysis tools. These tools come with their own conda environments and databases. It is quite likely that the users have already installed these tools on their servers. As a result, I have decided not to include everything in a big conda environment, but rather to keep each tool in its own conda environment, separated from the conda environment of DoViP.","category":"page"}]
}
