var documenterSearchIndex = {"docs":
[{"location":"installation_conda_dependencies/#DoViP-dependencies-and-their-installation","page":"Dependencies","title":"DoViP dependencies and their installation","text":"","category":"section"},{"location":"installation_conda_dependencies/#geNomad","page":"Dependencies","title":"geNomad","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"geNomad use and installation is described in detail here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"mamba create -p path/to/conda_genomad -c conda-forge -c bioconda genomad\n\n# database setup\nconda activate conda_genomad\ncd /path/for/databases\ngenomad download-database .","category":"page"},{"location":"installation_conda_dependencies/#VIBRANT","page":"Dependencies","title":"VIBRANT","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"VIBRANT use and installation is described in detail here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# env\nconda create -n conda_VIBRANT\nconda activate conda_VIBRANT\n\n# dependencies\nconda install -c bioconda hmmer\nconda install -c bioconda prodigal\nconda install -c bioconda biopython\nconda install -c anaconda pandas\nconda install matplotlib\nconda install seaborn\nconda install -c anaconda numpy\nconda install scikit-learn\n\n# install VIBRANT itself\ncd path/for/vibrant/repo\ngit clone https://github.com/AnantharamanLab/VIBRANT\nchmod -R 777 VIBRANT/\n\n# database\ncd databases\npip install --upgrade scikit-learn==0.21.3\npip install --upgrade numpy==1.17.0\npython3 VIBRANT_setup.py ","category":"page"},{"location":"installation_conda_dependencies/#VirSorter2","page":"Dependencies","title":"VirSorter2","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"VirSorter 2 use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# create env and instal, virsorter2\nconda create -p path/to/conda_virsorter2\nmamba activate path/to/conda_virsorter2\nmamba install -c conda-forge -c bioconda virsorter=2\n\n# databases\ncd conda_virsorter2/\nvirsorter setup -d db -j 4","category":"page"},{"location":"installation_conda_dependencies/#DeepVir-Finder","page":"Dependencies","title":"DeepVir Finder","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"DeepVir Finder use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# create env\nconda create -p /path/to/conda_DVF python=3.6 numpy theano=1.0.3 keras=2.2.4 scikit-learn Biopython h5py\nconda activate conda_DVF\n\n# clone the git repo\ncd path/for/repo\ngit clone https://github.com/jessieren/DeepVirFinder","category":"page"},{"location":"installation_conda_dependencies/#CheckV","page":"Dependencies","title":"CheckV","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"CheckV use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"# env\nconda create -p path/to/conda_checkV\nconda activate conda_checkV\n\n#\nconda install -c conda-forge -c bioconda checkV\n\n\n# database\ncd path/for/databases/\ncheckv download_database ./","category":"page"},{"location":"installation_conda_dependencies/#PhaBox","page":"Dependencies","title":"PhaBox","text":"","category":"section"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"PhaBox use and installation are detailed here. Below is the installation procedure we used on our servers.","category":"page"},{"location":"installation_conda_dependencies/","page":"Dependencies","title":"Dependencies","text":"cd path/for/repo\ngit clone https://github.com/KennthShang/PhaBOX\ncd PhaBOX/\nconda env create -f webserver.yml -p path/for/conda_PhaBOX\nconda activate conda_PhaBOX\n\n# databases\npip install gdown\ngdown  --id 1hjACPsIOqqcS5emGaduYvYrCzrIpt2_9\ngdown  --id 1E94ii3Q0O8ZBm7UsyDT_n06YekNtfV20\n\npath/for/repo/unzip phagesuite_database.zip  > /dev/null\npath/for/repo/unzip phagesuite_parameters.zip  > /dev/null\n\n\ncp blastxml_to_tabular.pypath/for/conda_PhaBOX/bin/blastxml_to_tabular.py\nchmod 777 path/for/conda_PhaBOX/bin/blastxml_to_tabular.py","category":"page"},{"location":"installation_DoViP_env/#Variant-1:-Step-by-step","page":"DoViP environment","title":"Variant 1: Step by step","text":"","category":"section"},{"location":"installation_DoViP_env/#1.-Create-the-conda-environment-for-DoViP-and-install-Julia-programming-language","page":"DoViP environment","title":"1. Create the conda environment for DoViP and install Julia programming language","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"# create conda env\nconda create -p /path/toward/conda_DoViP\nconda activate /path/toward/conda_DoViP\n\n# download and install julia version 1.10.0\ncd /path/toward/conda_DoViP\nwget https://julialang-s3.julialang.org/bin/linux/x64/1.10/julia-1.10.0-linux-x86_64.tar.gz\ntar -xzvf julia-1.10.0-linux-x86_64.tar.gz\nrm julia-1.10.0-linux-x86_64.tar.gz\nmkdir bin\n\n# make a symbolic link of the julia executable in the /path/toward/conda_DoViP/bin/ folder\nln -s /path/toward/conda_DoViP/julia-1.10.0/bin/julia /path/toward/conda_DoViP/bin/julia\n","category":"page"},{"location":"installation_DoViP_env/#2.-Bring-the-DoViP-repo","page":"DoViP environment","title":"2. Bring the DoViP repo","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"cd /path/toward/conda_DoViP\ngit clone https://github.com/CristinaMoraru/DoViP.git\n\nmv DoViP/* .\nrm DoViP\n\n# give permissions to the folder where julia saved its libraries\ncd share\nchmod 777 -R julia","category":"page"},{"location":"installation_DoViP_env/#3.-Refresh-the-DoViP-conda-environment","page":"DoViP environment","title":"3. Refresh the DoViP conda environment","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"conda deactivate conda_DoViP\nconda activate /path/toward/conda_DoViP\n","category":"page"},{"location":"installation_DoViP_env/#4.-Instantiate-DoViP-libraries","page":"DoViP environment","title":"4. Instantiate DoViP libraries","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"\ncd /path/toward/conda_DoViP/\n\njulia -e 'using Pkg; Pkg.activate(\"DoViP_App.jl\"); Pkg.instantiate()' \n","category":"page"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"Instantiation is necessary to bring on your machine the packages on which DoViP depends. After instantiation, it will attempt to start DoViP already, and therefore you will see an error message:  \"\". Ignore it.","category":"page"},{"location":"installation_DoViP_env/#5.Test-installation","page":"DoViP environment","title":"5.Test installation","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"DoViP --help","category":"page"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"You should get the following message: \"DoViP - a workflow for virus prediction in metagenomes.\"","category":"page"},{"location":"installation_DoViP_env/#From-a-conda-package","page":"DoViP environment","title":"From a conda package","text":"","category":"section"},{"location":"installation_DoViP_env/","page":"DoViP environment","title":"DoViP environment","text":"to be prepared","category":"page"},{"location":"Running/","page":"Running","title":"Running","text":"re","category":"page"},{"location":"#DoViP-Do-Virus-Prediction","page":"Home","title":"DoViP - Do Virus Prediction","text":"","category":"section"},{"location":"Installation/","page":"Overview","title":"Overview","text":"DoViP runs and then integrates the results from several virus analysis tools. These tools come with their own conda environments and databases. It is quite likely that the users have already installed these tools on their servers. As a result, I have decided not to include everything in a big conda environment, but rather to keep each tool in its own conda environment, separated from the conda environment of DoViP.","category":"page"}]
}
